// @shopify/polaris-viz/build/esm/components/LineChartPredictive/components/PredictiveLinePoints/PredictiveLinePoints.js
import { uniqueId, COLOR_VISION_SINGLE_ITEM, isGradientType, changeColorOpacity, LinearGradientWithStops, changeGradientOpacity } from '@shopify/polaris-viz-core';
import { useState, useMemo, Fragment } from 'react';
import { jsx, jsxs } from 'react/jsx-runtime';
import { useWatchColorVisionEvents } from '../../../../hooks/ColorVisionA11y/useWatchColorVisionEvents.js';
import { Point } from '../../../Point/Point.js';

function PredictiveLinePoints({
  data,
  xScale,
  yScale
}) {
  const [activeLineIndex, setActiveLineIndex] = useState(-1);
  const id = useMemo(() => uniqueId('PredictiveLinePoints'), []);
  useWatchColorVisionEvents({
    type: COLOR_VISION_SINGLE_ITEM,
    onIndexChange: ({
      detail
    }) => setActiveLineIndex(detail.index)
  });
  return /*#__PURE__*/jsx(Fragment, {
    children: data.map((series, seriesIndex) => {
      var _series$metadata, _series$metadata$rela, _series$metadata2, _series$data$predicti, _series$data$predicti2;

      if (((_series$metadata = series.metadata) === null || _series$metadata === void 0 ? void 0 : _series$metadata.isPredictive) == null) {
        return false;
      }

      const index = (_series$metadata$rela = (_series$metadata2 = series.metadata) === null || _series$metadata2 === void 0 ? void 0 : _series$metadata2.relatedIndex) !== null && _series$metadata$rela !== void 0 ? _series$metadata$rela : seriesIndex;
      const pointGradientId = `${id}-point-${index}`;
      const predictiveStartIndex = series.data.findIndex(({
        key
      }) => {
        var _series$metadata3;

        return key === ((_series$metadata3 = series.metadata) === null || _series$metadata3 === void 0 ? void 0 : _series$metadata3.startKey);
      });
      const color = series.color;
      const pointColor = isGradientType(color) ? `url(#${pointGradientId})` : changeColorOpacity(color);
      return /*#__PURE__*/jsxs(Fragment, {
        children: [isGradientType(color) ? /*#__PURE__*/jsx("defs", {
          children: /*#__PURE__*/jsx(LinearGradientWithStops, {
            id: pointGradientId,
            gradient: changeGradientOpacity(color),
            gradientUnits: "userSpaceOnUse",
            y1: "100%",
            y2: "0%"
          })
        }) : null, /*#__PURE__*/jsx(Point, {
          color: pointColor,
          cx: xScale(predictiveStartIndex),
          cy: yScale((_series$data$predicti = (_series$data$predicti2 = series.data[predictiveStartIndex]) === null || _series$data$predicti2 === void 0 ? void 0 : _series$data$predicti2.value) !== null && _series$data$predicti !== void 0 ? _series$data$predicti : -1),
          active: activeLineIndex === -1 || activeLineIndex === index,
          index: index,
          isAnimated: false,
          ariaHidden: true
        })]
      }, `${series.name}-${index}`);
    })
  });
}

export { PredictiveLinePoints };
