// @shopify/polaris-viz-core/build/cjs/hooks/useChartPositions.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var constants = require('../constants.js');
var useTheme = require('./useTheme.js');

function useChartPositions({
  annotationsHeight,
  height,
  width,
  xAxisHeight,
  yAxisWidth
}) {
  const {
    grid: {
      horizontalMargin
    }
  } = useTheme.useTheme();
  const chartXPosition = horizontalMargin + yAxisWidth + constants.Y_AXIS_CHART_SPACING;
  const chartYPosition = constants.ChartMargin.Top + annotationsHeight;
  const drawableHeight = height - chartYPosition - xAxisHeight - constants.LABEL_AREA_TOP_SPACING;
  const drawableWidth = width - chartXPosition - horizontalMargin;
  return {
    chartXPosition,
    chartYPosition,
    drawableHeight,
    drawableWidth,
    xAxisBounds: {
      x: chartXPosition,
      y: drawableHeight + constants.LABEL_AREA_TOP_SPACING + chartYPosition
    },
    yAxisBounds: {
      x: horizontalMargin,
      y: chartYPosition
    }
  };
}

exports.useChartPositions = useChartPositions;
