// @shopify/polaris-viz/build/cjs/hooks/useHorizontalTransitions.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var web = require('@react-spring/web');
var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');

// when returning values from a function. It's fixed
// in v9.2.5-beta.0 but until it's stable
// we're going to cast.
// https://github.com/pmndrs/react-spring/issues/1483

function useHorizontalTransitions({
  chartXPosition,
  groupHeight,
  series
}) {
  const {
    shouldAnimate,
    id
  } = polarisVizCore.useChartContext();
  const seriesWithIndex = React.useMemo(() => {
    return series[0].data.map(({
      key
    }, index) => ({
      key: `${key}`,
      index
    }));
  }, [series]);

  const getTransform = index => {
    return `translate(${chartXPosition}px,${groupHeight * index}px)`;
  };

  const animationTrail = 50;
  const outOfChartPosition = getTransform(series.length + 1);
  const transitions = web.useTransition(seriesWithIndex, {
    keys: item => {
      return item.key === '' ? `${id}-${item.index}` : item.key;
    },
    initial: ({
      index
    }) => ({
      opacity: 1,
      transform: getTransform(index)
    }),
    from: ({
      index
    }) => ({
      opacity: 0,
      transform: shouldAnimate ? outOfChartPosition : getTransform(index)
    }),
    leave: {
      opacity: 0,
      transform: outOfChartPosition
    },
    update: ({
      index
    }) => ({
      opacity: 1,
      transform: getTransform(index)
    }),
    expires: true,
    config: polarisVizCore.BARS_SORT_TRANSITION_CONFIG,
    trail: shouldAnimate ? animationTrail : 0,
    immediate: !shouldAnimate
  });
  return {
    transitions
  };
}

exports.useHorizontalTransitions = useHorizontalTransitions;
