// @shopify/polaris-viz/build/cjs/components/VerticalBarChart/components/StackedBarGroups/StackedBarGroups.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var getLoadAnimationDelay = require('../../../../utilities/getLoadAnimationDelay.js');
var StackedBarGroups$1 = require('./StackedBarGroups.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var useStackedGapsForVerticalChart = require('../../hooks/useStackedGapsForVerticalChart.js');
var formatAriaLabel = require('../../utilities/formatAriaLabel.js');
var Stack = require('./components/Stack/Stack.js');

function StackedBarGroups({
  accessibilityData,
  activeBarGroup,
  drawableHeight,
  gapWidth,
  id,
  labels,
  stackedValues,
  xScale,
  yScale
}) {
  const width = xScale.bandwidth() - polarisVizCore.BAR_SPACING;
  const formattedStackedValues = React.useMemo(() => {
    return labels.map((_, labelIndex) => {
      return stackedValues.map(data => {
        return data[labelIndex];
      });
    });
  }, [labels, stackedValues]);
  const gaps = useStackedGapsForVerticalChart.useStackedGapsForVerticalChart({
    stackedValues,
    labels
  });
  return /*#__PURE__*/jsxRuntime.jsx(React.Fragment, {
    children: formattedStackedValues.map((item, groupIndex) => {
      var _xScale;

      const x = (_xScale = xScale(groupIndex.toString())) !== null && _xScale !== void 0 ? _xScale : 0;
      const groupAriaLabel = formatAriaLabel.formatAriaLabel(accessibilityData[groupIndex]);
      const animationDelay = getLoadAnimationDelay.getLoadAnimationDelay(groupIndex, formattedStackedValues.length);
      return /*#__PURE__*/jsxRuntime.jsxs("g", { ...polarisVizCore.getColorVisionEventAttrs({
          type: polarisVizCore.COLOR_VISION_GROUP_ITEM,
          index: groupIndex
        }),
        className: StackedBarGroups$1["default"].Group,
        style: polarisVizCore.getColorVisionStylesForActiveIndex({
          activeIndex: activeBarGroup,
          index: groupIndex
        }),
        "aria-label": groupAriaLabel,
        role: "list",
        "data-type": polarisVizCore.DataType.BarGroup,
        "data-index": groupIndex,
        "aria-hidden": "false",
        children: [/*#__PURE__*/jsxRuntime.jsx("rect", {
          height: drawableHeight,
          fill: "transparent",
          x: x - gapWidth / 2,
          width: xScale.bandwidth() + gapWidth,
          "aria-hidden": "true"
        }), /*#__PURE__*/jsxRuntime.jsx(Stack.Stack, {
          activeBarGroup: activeBarGroup,
          animationDelay: animationDelay,
          data: item,
          gaps: gaps,
          groupIndex: groupIndex,
          id: id,
          width: width,
          x: x,
          yScale: yScale
        })]
      }, groupIndex);
    })
  });
}

exports.StackedBarGroups = StackedBarGroups;
