// @shopify/polaris-viz/build/cjs/components/ComboChart/hooks/useDualAxisScale.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var getSecondaryDataForMixedData = require('../utilities/getSecondaryDataForMixedData.js');

function useDualAxisScale({
  doBothChartsContainMixedValues,
  doesOneChartContainAllNegativeValues,
  drawableHeight,
  primaryAxis,
  secondaryAxis,
  shouldPlaceZeroInMiddleOfChart,
  ticksBetweenZeroAndMax,
  yScale
}) {
  const {
    secondaryDrawableHeight,
    secondaryMax,
    secondaryMin
  } = getSecondaryDataForMixedData.getSecondaryDataForMixedData({
    doBothChartsContainMixedValues,
    doesOneChartContainAllNegativeValues,
    drawableHeight,
    secondaryAxis,
    shouldPlaceZeroInMiddleOfChart,
    yScale
  });
  const {
    yScale: secondaryYScale
  } = polarisVizCore.useYScale({
    drawableHeight: secondaryDrawableHeight,
    formatYAxisLabel: secondaryAxis.yAxisOptions.labelFormatter,
    integersOnly: secondaryAxis.yAxisOptions.integersOnly,
    max: polarisVizCore.getClosestDivisibleNumber(secondaryMax, ticksBetweenZeroAndMax),
    min: secondaryMin,
    // For non-source of truth, the ticks are exactly
    // what they should be, so we don't want to apply .nice()
    // to the scale.
    shouldRoundUp: false
  });
  const barYScale = primaryAxis.shape === 'Bar' ? yScale : secondaryYScale;
  const lineYScale = primaryAxis.shape === 'Bar' ? secondaryYScale : yScale;
  return {
    barYScale,
    lineYScale,
    primaryYScale: yScale,
    secondaryYScale
  };
}

exports.useDualAxisScale = useDualAxisScale;
