// @shopify/polaris-viz/build/cjs/components/TooltipContent/components/TooltipContentContainer/TooltipContentContainer.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var React = require('react');
var constants = require('../../../../constants.js');
var useBrowserCheck = require('../../../../hooks/useBrowserCheck.js');
var TooltipContentContainer$1 = require('./TooltipContentContainer.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var useWatchColorVisionEvents = require('../../../../hooks/ColorVisionA11y/useWatchColorVisionEvents.js');

function TooltipContentContainer({
  children,
  maxWidth,
  theme
}) {
  const {
    isFirefox
  } = useBrowserCheck.useBrowserCheck();
  const selectedTheme = polarisVizCore.useTheme(theme);
  const [activeColorVisionIndex, setActiveIndex] = React.useState(-1);
  useWatchColorVisionEvents.useWatchColorVisionEvents({
    type: polarisVizCore.COLOR_VISION_SINGLE_ITEM,
    onIndexChange: ({
      detail
    }) => setActiveIndex(detail.index)
  });
  return /*#__PURE__*/jsxRuntime.jsx("div", {
    className: TooltipContentContainer$1["default"].Container,
    style: {
      background: polarisVizCore.changeColorOpacity(selectedTheme.tooltip.backgroundColor, isFirefox ? 1 : constants.TOOLTIP_BG_OPACITY),
      maxWidth,
      fontFamily: polarisVizCore.FONT_FAMILY
    },
    children: children({
      activeColorVisionIndex
    })
  });
}

exports.TooltipContentContainer = TooltipContentContainer;
