// @shopify/polaris-viz/build/cjs/components/LineChart/LineChart.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var getTooltipContentRenderer = require('../../utilities/getTooltipContentRenderer.js');
var fillMissingDataPoints = require('../../utilities/fillMissingDataPoints.js');
var getLineChartDataWithDefaults = require('../../utilities/getLineChartDataWithDefaults.js');
var useThemeSeriesColors = require('../../hooks/useThemeSeriesColors.js');
var Chart = require('./Chart.js');
var jsxRuntime = require('react/jsx-runtime');
var getAxisOptions = require('../../utilities/getAxisOptions.js');
var normalizeData = require('../../utilities/normalizeData.js');
var SkipLink = require('../SkipLink/SkipLink.js');
var ChartContainer = require('../ChartContainer/ChartContainer.js');
var ChartSkeleton = require('../ChartSkeleton/ChartSkeleton.js');

function LineChart(props) {
  const {
    defaultTheme
  } = polarisVizCore.usePolarisVizContext();
  const {
    annotations = [],
    data: dataSeries,
    emptyStateText,
    errorText,
    id,
    isAnimated,
    onError,
    renderLegendContent,
    renderHiddenLegendLabel,
    seriesNameFormatter = value => `${value}`,
    showLegend = true,
    hideLegendOverflow = true,
    skipLinkText,
    state,
    theme = defaultTheme,
    tooltipOptions,
    xAxisOptions,
    yAxisOptions
  } = { ...polarisVizCore.DEFAULT_CHART_PROPS,
    ...props
  };
  const data = fillMissingDataPoints.fillMissingDataPoints(dataSeries, true);
  const selectedTheme = polarisVizCore.useTheme(theme);
  const seriesColors = useThemeSeriesColors.useThemeSeriesColors(data, selectedTheme);
  const skipLinkAnchorId = React.useRef(polarisVizCore.uniqueId('lineChart'));
  const xAxisOptionsWithDefaults = getAxisOptions.getXAxisOptionsWithDefaults(xAxisOptions);
  const yAxisOptionsWithDefaults = getAxisOptions.getYAxisOptionsWithDefaults(yAxisOptions);
  const renderTooltip = getTooltipContentRenderer.getTooltipContentRenderer({
    tooltipOptions,
    theme,
    data
  });
  const annotationsLookupTable = normalizeData.normalizeData(annotations, 'startKey');
  const dataWithDefaults = getLineChartDataWithDefaults.getLineChartDataWithDefaults(data, seriesColors);
  return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
    children: [skipLinkText == null || skipLinkText.length === 0 || data.length === 0 ? null : /*#__PURE__*/jsxRuntime.jsx(SkipLink.SkipLink, {
      anchorId: skipLinkAnchorId.current,
      children: skipLinkText
    }), /*#__PURE__*/jsxRuntime.jsx(ChartContainer.ChartContainer, {
      id: id,
      data: data,
      theme: theme,
      isAnimated: isAnimated,
      type: polarisVizCore.InternalChartType.Line,
      onError: onError,
      children: state !== polarisVizCore.ChartState.Success ? /*#__PURE__*/jsxRuntime.jsx(ChartSkeleton.ChartSkeleton, {
        state: state,
        errorText: errorText,
        theme: theme
      }) : /*#__PURE__*/jsxRuntime.jsx(Chart.Chart, {
        annotationsLookupTable: annotationsLookupTable,
        data: dataWithDefaults,
        emptyStateText: emptyStateText,
        renderLegendContent: renderLegendContent,
        renderTooltipContent: renderTooltip,
        renderHiddenLegendLabel: renderHiddenLegendLabel,
        seriesNameFormatter: seriesNameFormatter,
        showLegend: showLegend,
        hideLegendOverflow: hideLegendOverflow,
        theme: theme,
        xAxisOptions: xAxisOptionsWithDefaults,
        yAxisOptions: yAxisOptionsWithDefaults,
        slots: props.slots
      })
    }), skipLinkText == null || skipLinkText.length === 0 ? null : /*#__PURE__*/jsxRuntime.jsx(SkipLink.SkipLink.Anchor, {
      id: skipLinkAnchorId.current
    })]
  });
}

exports.LineChart = LineChart;
