// @shopify/polaris-viz/build/cjs/components/SimpleBarChart/SimpleBarChart.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var Chart = require('./Chart.js');
var jsxRuntime = require('react/jsx-runtime');
var getAxisOptions = require('../../utilities/getAxisOptions.js');
var ChartContainer = require('../ChartContainer/ChartContainer.js');
var ChartSkeleton = require('../ChartSkeleton/ChartSkeleton.js');

function SimpleBarChart(props) {
  const {
    defaultTheme
  } = polarisVizCore.usePolarisVizContext();
  const {
    id,
    isAnimated,
    data,
    renderLegendContent,
    legendPosition = 'bottom-right',
    onError,
    seriesNameFormatter = value => `${value}`,
    showLegend = true,
    theme = defaultTheme,
    type = 'default',
    xAxisOptions,
    yAxisOptions,
    state,
    errorText
  } = { ...polarisVizCore.DEFAULT_CHART_PROPS,
    ...props
  };
  const xAxisOptionsWithDefaults = getAxisOptions.getXAxisOptionsWithDefaults(xAxisOptions);
  const yAxisOptionsWithDefaults = getAxisOptions.getYAxisOptionsWithDefaults(yAxisOptions);
  return /*#__PURE__*/jsxRuntime.jsx(ChartContainer.ChartContainer, {
    data: data,
    theme: theme,
    id: id,
    isAnimated: isAnimated,
    onError: onError,
    children: state !== polarisVizCore.ChartState.Success ? /*#__PURE__*/jsxRuntime.jsx(ChartSkeleton.ChartSkeleton, {
      type: "SimpleBar",
      state: state,
      errorText: errorText,
      theme: theme
    }) : /*#__PURE__*/jsxRuntime.jsx(Chart.Chart, {
      data: data,
      renderLegendContent: renderLegendContent,
      seriesNameFormatter: seriesNameFormatter,
      legendPosition: legendPosition,
      showLegend: showLegend,
      type: type,
      xAxisOptions: xAxisOptionsWithDefaults,
      yAxisOptions: yAxisOptionsWithDefaults
    })
  });
}

exports.SimpleBarChart = SimpleBarChart;
