// @shopify/polaris-viz/build/esm/components/SparkBarChart/Chart.js
import { Fragment } from 'react';
import { useTransition } from '@react-spring/web';
import { useChartContext, XMLNS, ANIMATION_MARGIN, SparkBarSeries } from '@shopify/polaris-viz-core';
import styles from './SparkBarChart.scss.js';
import { jsxs, jsx } from 'react/jsx-runtime';

function Chart({
  data,
  dimensions,
  targetLine,
  accessibilityLabel
}) {
  const {
    shouldAnimate
  } = useChartContext();
  const {
    width,
    height
  } = dimensions !== null && dimensions !== void 0 ? dimensions : {
    width: 0,
    height: 0
  };
  const viewboxHeight = height + ANIMATION_MARGIN * 2;
  return /*#__PURE__*/jsxs(Fragment, {
    children: [accessibilityLabel ? /*#__PURE__*/jsx("span", {
      className: styles.VisuallyHidden,
      children: accessibilityLabel
    }) : null, /*#__PURE__*/jsx("svg", {
      xmlns: XMLNS,
      "aria-hidden": true,
      viewBox: `0 -${ANIMATION_MARGIN} ${width} ${viewboxHeight}`,
      style: {
        transform: `translateY(-${ANIMATION_MARGIN}px)`
      },
      className: styles.Svg,
      height: viewboxHeight,
      width: width,
      children: /*#__PURE__*/jsx(SparkBarSeries, {
        data: data,
        targetLine: targetLine,
        height: height,
        shouldAnimate: shouldAnimate,
        useTransition: useTransition,
        width: width
      })
    })]
  });
}

export { Chart };
