// @shopify/polaris-viz/build/esm/components/SparkBarChart/SparkBarChart.js
import { usePolarisVizContext, DEFAULT_CHART_PROPS, ChartState } from '@shopify/polaris-viz-core';
import { Chart } from './Chart.js';
import { jsx } from 'react/jsx-runtime';
import { ChartContainer } from '../ChartContainer/ChartContainer.js';
import { ChartSkeleton } from '../ChartSkeleton/ChartSkeleton.js';

function SparkBarChart(props) {
  const {
    defaultTheme
  } = usePolarisVizContext();
  const {
    data,
    accessibilityLabel,
    id,
    isAnimated,
    onError,
    targetLine,
    theme = defaultTheme,
    state,
    errorText
  } = { ...DEFAULT_CHART_PROPS,
    ...props
  };
  return /*#__PURE__*/jsx(ChartContainer, {
    data: data,
    theme: theme,
    sparkChart: true,
    id: id,
    isAnimated: isAnimated,
    onError: onError,
    children: state !== ChartState.Success ? /*#__PURE__*/jsx(ChartSkeleton, {
      type: "Spark",
      state: state,
      errorText: errorText,
      theme: theme
    }) : /*#__PURE__*/jsx(Chart, {
      data: data,
      targetLine: targetLine,
      accessibilityLabel: accessibilityLabel,
      theme: theme
    })
  });
}

export { SparkBarChart };
