// @shopify/polaris-viz/build/cjs/components/FunnelChart/components/FunnelChartXAxisArrows.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var useLabels = require('../../Labels/hooks/useLabels.js');
var TextLine = require('../../TextLine/TextLine.js');

const ARROW_WIDTH = 11;
const ARROW_HEIGHT = 9;
function FunnelChartXAxisArrows({
  onHeightChange,
  x,
  index,
  chartX,
  chartY,
  labelWidth
}) {
  var _firstLabel$transform;

  const {
    lines
  } = useLabels.useLabels({
    allowLineWrap: true,
    labels: ['→'],
    targetWidth: labelWidth,
    onHeightChange
  });
  const firstLine = lines[0];
  const firstLabel = firstLine[0];
  const areLabelsVertical = (_firstLabel$transform = firstLabel.transform) === null || _firstLabel$transform === void 0 ? void 0 : _firstLabel$transform.includes('rotate(-90)');

  const getArrowTransform = () => {
    if (areLabelsVertical) {
      return `translate(${chartX + x - labelWidth / 2 + ARROW_WIDTH / 2},${chartY + ARROW_HEIGHT})`;
    }

    return `translate(${chartX - labelWidth / 1.5 + x},${chartY})`;
  };

  const arrowLine = [{ ...firstLabel,
    transform: undefined
  }];
  return /*#__PURE__*/jsxRuntime.jsx("g", {
    transform: getArrowTransform(),
    children: /*#__PURE__*/jsxRuntime.jsx(TextLine.TextLine, {
      line: arrowLine,
      index: index
    })
  }, `${index}-arrow`);
}

exports.FunnelChartXAxisArrows = FunnelChartXAxisArrows;
