// @shopify/polaris-viz/build/cjs/components/TextLine/TextLine.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var jsxRuntime = require('react/jsx-runtime');

function TextLine({
  index,
  line
}) {
  const selectedTheme = polarisVizCore.useTheme();
  return /*#__PURE__*/jsxRuntime.jsx(React.Fragment, {
    children: line.map(({
      dominantBaseline,
      height,
      fullText,
      truncatedText,
      textAnchor,
      transform,
      width,
      x,
      y
    }, textIndex) => {
      const key = `${index}-${textIndex}-${truncatedText}`;
      return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
        children: [/*#__PURE__*/jsxRuntime.jsx("text", {
          textAnchor: textAnchor,
          dominantBaseline: dominantBaseline,
          height: height,
          width: width,
          x: x,
          y: y,
          fill: selectedTheme.xAxis.labelColor,
          fontSize: polarisVizCore.FONT_SIZE,
          fontFamily: polarisVizCore.FONT_FAMILY,
          transform: transform,
          children: truncatedText
        }), /*#__PURE__*/jsxRuntime.jsx("title", {
          children: fullText
        })]
      }, key);
    })
  });
}

exports.TextLine = TextLine;
