// @shopify/polaris-viz/build/esm/components/LineChartRelational/components/Area/Area.js
import { useSpring, animated } from '@react-spring/web';
import { usePrevious, getColorVisionStylesForActiveIndex } from '@shopify/polaris-viz-core';
import { jsx } from 'react/jsx-runtime';

function Area({
  activeIndex,
  fill,
  getAreaGenerator,
  hiddenIndexes = [],
  index,
  series,
  shouldAnimate = false
}) {
  var _series$metadata$rela, _series$metadata;

  const pathD = getAreaGenerator(series);
  const previous = usePrevious(pathD);
  const spring = useSpring({
    from: {
      pathD: previous
    },
    to: {
      pathD
    },
    immediate: !shouldAnimate
  });

  if (hiddenIndexes.includes(index)) {
    return null;
  }

  if (hiddenIndexes.includes((_series$metadata$rela = (_series$metadata = series.metadata) === null || _series$metadata === void 0 ? void 0 : _series$metadata.relatedIndex) !== null && _series$metadata$rela !== void 0 ? _series$metadata$rela : -1)) {
    return null;
  }

  if (pathD == null) {
    return null;
  }

  return /*#__PURE__*/jsx(animated.path, {
    d: spring.pathD,
    fill: fill,
    style: { ...getColorVisionStylesForActiveIndex({
        activeIndex,
        index,
        fadedOpacity: 0.2
      })
    }
  });
}

export { Area };
