// @shopify/polaris-viz/build/cjs/components/LineChartRelational/components/MissingDataArea/MissingDataArea.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var groupNumbersIntoRuns = require('./utilities/groupNumbersIntoRuns.js');
var jsxRuntime = require('react/jsx-runtime');

function MissingDataAreaRaw({
  data,
  drawableHeight,
  xScale
}) {
  const selectedTheme = polarisVizCore.useTheme();
  const patternID = React.useRef(polarisVizCore.uniqueId('missingDataPattern'));
  const dataSeriesHasNulls = React.useRef(false);
  const dataLength = data[0].data.length - 1;
  const indexesWithData = new Set([]);

  for (const series of data) {
    const hasAnyNulls = series.data.some(({
      value
    }) => value == null);

    if (!hasAnyNulls) {
      continue;
    }

    for (const [index, {
      value
    }] of series.data.entries()) {
      if (value != null) {
        indexesWithData.add(index);
      } else {
        dataSeriesHasNulls.current = true;
      }
    }
  }

  if (indexesWithData.size === 0 || dataSeriesHasNulls.current === false) {
    return null;
  }

  const sortedIndexes = [...indexesWithData].sort((current, next) => current - next);
  const groups = groupNumbersIntoRuns.groupNumbersIntoRuns(sortedIndexes);
  return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
    children: [/*#__PURE__*/jsxRuntime.jsxs("defs", {
      children: [/*#__PURE__*/jsxRuntime.jsx("pattern", {
        id: patternID.current,
        patternUnits: "userSpaceOnUse",
        width: "12",
        height: "12",
        patternTransform: "rotate(135)",
        children: /*#__PURE__*/jsxRuntime.jsx("line", {
          x1: "0",
          y: "0",
          x2: "0",
          y2: "12",
          stroke: selectedTheme.missingData.lineColor,
          strokeWidth: "12",
          opacity: "0.2"
        })
      }), /*#__PURE__*/jsxRuntime.jsxs("mask", {
        id: `${patternID.current}-clip`,
        children: [/*#__PURE__*/jsxRuntime.jsx("rect", {
          x: xScale(0),
          y: 0,
          height: drawableHeight,
          width: xScale(dataLength),
          fill: "white"
        }), groups.map((indexes, index) => {
          const startIndex = Math.min(...indexes);
          const endIndex = Math.max(...indexes);
          const width = xScale(endIndex - startIndex);
          return /*#__PURE__*/jsxRuntime.jsx("rect", {
            x: xScale(startIndex),
            y: 0,
            height: drawableHeight,
            width: width,
            fill: "black"
          }, index);
        })]
      })]
    }), /*#__PURE__*/jsxRuntime.jsx("rect", {
      x: xScale(0),
      y: 0,
      height: drawableHeight,
      width: xScale(dataLength),
      fill: `url(#${patternID.current})`,
      mask: `url(#${patternID.current}-clip)`
    })]
  });
}

const MissingDataArea = /*#__PURE__*/React.memo(MissingDataAreaRaw);

exports.MissingDataArea = MissingDataArea;
