// @shopify/polaris-viz/build/cjs/components/VerticalBarChart/VerticalBarChart.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Chart = require('./Chart.js');
var jsxRuntime = require('react/jsx-runtime');
var polarisVizCore = require('@shopify/polaris-viz-core');

function VerticalBarChart({
  annotationsLookupTable = {},
  data,
  dimensions,
  emptyStateText,
  renderLegendContent,
  renderTooltipContent,
  showLegend,
  type = 'default',
  xAxisOptions,
  yAxisOptions,
  renderHiddenLegendLabel,
  seriesNameFormatter
}) {
  const selectedTheme = polarisVizCore.useTheme();
  const seriesColors = polarisVizCore.useThemeSeriesColors(data, selectedTheme);
  const seriesWithDefaults = data.map((series, index) => ({
    color: seriesColors[index],
    ...series
  }));
  return /*#__PURE__*/jsxRuntime.jsx(Chart.Chart, {
    dimensions: dimensions,
    annotationsLookupTable: annotationsLookupTable,
    data: seriesWithDefaults,
    emptyStateText: emptyStateText,
    renderLegendContent: renderLegendContent,
    renderTooltipContent: renderTooltipContent,
    seriesNameFormatter: seriesNameFormatter,
    showLegend: showLegend,
    type: type,
    xAxisOptions: xAxisOptions,
    yAxisOptions: yAxisOptions,
    renderHiddenLegendLabel: renderHiddenLegendLabel
  });
}

exports.VerticalBarChart = VerticalBarChart;
