// @shopify/polaris-viz/build/cjs/components/shared/HorizontalBars/HorizontalBars.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var getTrendIndicatorData = require('../../../utilities/getTrendIndicatorData.js');
var HorizontalBars$1 = require('./HorizontalBars.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var useWatchColorVisionEvents = require('../../../hooks/ColorVisionA11y/useWatchColorVisionEvents.js');
var getHoverZoneOffset = require('../../../utilities/getHoverZoneOffset.js');
var Bar = require('../Bar/Bar.js');
var GradientDefs = require('../GradientDefs/GradientDefs.js');
var LabelWrapper = require('../LabelWrapper/LabelWrapper.js');
var Label = require('../Label/Label.js');
var TrendIndicator = require('../../TrendIndicator/TrendIndicator.js');
var constants = require('../../TrendIndicator/constants.js');

const SERIES_DELAY = 150;
function HorizontalBars({
  activeGroupIndex,
  animationDelay = 0,
  barHeight,
  data,
  groupIndex,
  id,
  isSimple,
  labelFormatter,
  name,
  xScale,
  zeroPosition,
  containerWidth,
  areAllNegative
}) {
  const selectedTheme = polarisVizCore.useTheme();
  const {
    characterWidths,
    theme
  } = polarisVizCore.useChartContext();
  const [activeBarIndex, setActiveBarIndex] = React.useState(-1);
  useWatchColorVisionEvents.useWatchColorVisionEvents({
    type: polarisVizCore.COLOR_VISION_SINGLE_ITEM,
    onIndexChange: ({
      detail
    }) => {
      if (activeGroupIndex === -1 || activeGroupIndex === groupIndex) {
        setActiveBarIndex(detail.index);
      }
    }
  });
  return /*#__PURE__*/jsxRuntime.jsx("g", {
    transform: `translate(${zeroPosition},${polarisVizCore.HORIZONTAL_GROUP_LABEL_HEIGHT})`,
    "aria-hidden": "true",
    children: data.map((_, seriesIndex) => {
      var _data$seriesIndex, _data$seriesIndex$met;

      if (data[seriesIndex].data[groupIndex] == null) {
        return;
      }

      const seriesAnimationDelay = animationDelay + SERIES_DELAY * seriesIndex;
      const {
        value
      } = data[seriesIndex].data[groupIndex];

      if (value == null) {
        return null;
      }

      const isNegative = value && value < 0;
      const label = labelFormatter(value);
      const ariaLabel = `${data[seriesIndex] ? data[seriesIndex].name : ''} ${value}`;
      const {
        trendIndicatorProps,
        trendIndicatorWidth
      } = getTrendIndicatorData.getTrendIndicatorData((_data$seriesIndex = data[seriesIndex]) === null || _data$seriesIndex === void 0 ? void 0 : (_data$seriesIndex$met = _data$seriesIndex.metadata) === null || _data$seriesIndex$met === void 0 ? void 0 : _data$seriesIndex$met.trends[groupIndex]);
      const labelWidth = polarisVizCore.estimateStringWidth(`${label}`, characterWidths);

      function getBarWidthAndLabelX() {
        const width = Math.abs(xScale(value !== null && value !== void 0 ? value : 0) - xScale(0));
        const itemSpacing = trendIndicatorProps == null ? polarisVizCore.HORIZONTAL_BAR_LABEL_OFFSET : polarisVizCore.HORIZONTAL_BAR_LABEL_OFFSET * 2;
        const leftLabelOffset = isSimple ? labelWidth + itemSpacing + trendIndicatorWidth : 0;
        const clampedWidth = polarisVizCore.clamp({
          amount: width,
          min: 1,
          max: Infinity
        });

        if (isNegative) {
          return {
            labelX: -(clampedWidth + leftLabelOffset),
            barWidth: clampedWidth
          };
        }

        return {
          labelX: clampedWidth + polarisVizCore.HORIZONTAL_BAR_LABEL_OFFSET,
          barWidth: clampedWidth
        };
      }

      const {
        labelX,
        barWidth
      } = getBarWidthAndLabelX();
      const y = barHeight * seriesIndex + polarisVizCore.HORIZONTAL_SPACE_BETWEEN_SINGLE * seriesIndex;
      const {
        clampedSize
      } = getHoverZoneOffset.getHoverZoneOffset({
        barSize: barWidth,
        zeroPosition: xScale(0),
        max: containerWidth,
        position: 'horizontal'
      });
      const trendYOffset = (barHeight - constants.HEIGHT) / 2;
      return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
        children: [/*#__PURE__*/jsxRuntime.jsx(Bar.Bar, {
          animationDelay: seriesAnimationDelay,
          areAllNegative: areAllNegative,
          color: `url(#${GradientDefs.getGradientDefId(theme, seriesIndex, id)})`,
          height: barHeight,
          index: groupIndex,
          isActive: activeBarIndex === -1 || activeBarIndex === seriesIndex,
          transform: isNegative ? 'scaleX(-1)' : '',
          width: barWidth,
          x: 0,
          y: y
        }), isSimple && /*#__PURE__*/jsxRuntime.jsxs(LabelWrapper.LabelWrapper, {
          animationDelay: seriesAnimationDelay,
          x: labelX,
          children: [/*#__PURE__*/jsxRuntime.jsx(Label.Label, {
            barHeight: barHeight,
            color: selectedTheme.xAxis.labelColor,
            label: label,
            labelWidth: labelWidth,
            y: y
          }), trendIndicatorProps != null && /*#__PURE__*/jsxRuntime.jsx("g", {
            transform: `translate(${labelWidth + polarisVizCore.HORIZONTAL_BAR_LABEL_OFFSET}, ${y + trendYOffset})`,
            children: /*#__PURE__*/jsxRuntime.jsx(TrendIndicator.TrendIndicator, { ...trendIndicatorProps
            })
          })]
        }), /*#__PURE__*/jsxRuntime.jsx("rect", {
          className: HorizontalBars$1["default"].Bar,
          x: 0,
          y: y - polarisVizCore.HORIZONTAL_SPACE_BETWEEN_SINGLE / 2,
          width: clampedSize,
          height: barHeight + polarisVizCore.HORIZONTAL_SPACE_BETWEEN_SINGLE,
          fill: "transparent",
          style: {
            transform: isNegative ? 'scaleX(-1)' : ''
          },
          ...polarisVizCore.getColorVisionEventAttrs({
            type: polarisVizCore.COLOR_VISION_SINGLE_ITEM,
            index: seriesIndex
          }),
          tabIndex: -1,
          role: "img",
          "aria-label": ariaLabel
        })]
      }, `series-${seriesIndex}-${id}-${name}`);
    })
  });
}

exports.HorizontalBars = HorizontalBars;
