// @shopify/polaris-viz/build/esm/components/LineChartPredictive/components/CustomLegend/CustomLegend.js
import styles from './CustomLegend.scss.js';
import { jsx } from 'react/jsx-runtime';
import { LegendItem } from '../../../Legend/components/LegendItem/LegendItem.js';
import { SeriesIcon } from '../SeriesIcon/SeriesIcon.js';

function CustomLegend({
  data,
  predictiveSeriesNames,
  getColorVisionEventAttrs,
  getColorVisionStyles,
  seriesNameFormatter,
  theme
}) {
  return /*#__PURE__*/jsx("ul", {
    className: styles.Container,
    children: data.map(({
      color,
      name,
      isComparison,
      metadata
    }, index) => {
      if (metadata !== null && metadata !== void 0 && metadata.isPredictive) {
        return null;
      }

      function renderSeriesIcon() {
        return /*#__PURE__*/jsx(SeriesIcon, {
          color: color
        });
      }

      return /*#__PURE__*/jsx("li", {
        style: { ...getColorVisionStyles(index)
        },
        ...getColorVisionEventAttrs(index),
        children: /*#__PURE__*/jsx(LegendItem, {
          color: color,
          index: index,
          renderSeriesIcon: predictiveSeriesNames.includes(name !== null && name !== void 0 ? name : '') ? renderSeriesIcon : undefined,
          isComparison: isComparison,
          name: seriesNameFormatter(name !== null && name !== void 0 ? name : ''),
          shape: "Line",
          theme: theme
        })
      }, index);
    })
  });
}

export { CustomLegend };
