// @shopify/polaris-viz/build/cjs/components/ChartSkeleton/components/SimpleNormalizedSkeleton/SimpleNormalizedSkeleton.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var SimpleNormalizedSkeleton$1 = require('./SimpleNormalizedSkeleton.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var ErrorText = require('../ErrorText/ErrorText.js');

const SIZE_TO_PX = {
  small: 16,
  medium: 36,
  large: 56
};
function SimpleNormalizedSkeleton({
  dimensions,
  errorText,
  showLegend,
  size,
  state
}) {
  const {
    width,
    height
  } = dimensions;
  const {
    grid: {
      color: gridColor
    },
    bar: {
      borderRadius
    },
    chartContainer: {
      padding
    }
  } = polarisVizCore.useTheme();
  const id = polarisVizCore.useUniqueId('simple-bar-skeleton');
  return /*#__PURE__*/jsxRuntime.jsxs("div", {
    style: {
      padding
    },
    children: [state === polarisVizCore.ChartState.Loading && /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
      children: [showLegend && /*#__PURE__*/jsxRuntime.jsx("div", {
        className: SimpleNormalizedSkeleton$1["default"].Legend,
        children: new Array(3).fill(0).map((_, index) => /*#__PURE__*/jsxRuntime.jsx("div", {
          className: SimpleNormalizedSkeleton$1["default"].LegendItem,
          children: new Array(3).fill(0).map((_, innerIndex) => /*#__PURE__*/jsxRuntime.jsx("div", {
            className: SimpleNormalizedSkeleton$1["default"].LegendItemComponent,
            style: {
              background: gridColor
            }
          }, `${id}${index}${innerIndex}`))
        }, `${id}${index}`))
      }), /*#__PURE__*/jsxRuntime.jsx("div", {
        style: {
          background: gridColor,
          borderRadius,
          height: SIZE_TO_PX[size]
        }
      })]
    }), state === polarisVizCore.ChartState.Error && /*#__PURE__*/jsxRuntime.jsx("svg", {
      viewBox: `0 0 ${width} ${height}`,
      children: /*#__PURE__*/jsxRuntime.jsx(ErrorText.ErrorText, {
        errorText: errorText,
        width: width,
        height: height
      })
    })]
  });
}

exports.SimpleNormalizedSkeleton = SimpleNormalizedSkeleton;
