// @shopify/polaris-viz/build/cjs/utilities/getTooltipContentRenderer.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var jsxRuntime = require('react/jsx-runtime');
var formatDataForTooltip = require('./formatDataForTooltip.js');
var TooltipContent = require('../components/TooltipContent/TooltipContent.js');

function getTooltipContentRenderer({
  tooltipOptions = {},
  theme,
  data
}) {
  return function renderTooltipContent(tooltipData) {
    if ((tooltipOptions === null || tooltipOptions === void 0 ? void 0 : tooltipOptions.renderTooltipContent) != null) {
      const {
        renderTooltipContent,
        ...formatters
      } = tooltipOptions;
      return renderTooltipContent({ ...tooltipData,
        dataSeries: data,
        formatters,
        theme
      });
    }

    const {
      title,
      formattedData
    } = formatDataForTooltip.formatDataForTooltip({
      data: tooltipData,
      tooltipOptions
    });

    if (formattedData[0].data.length === 0) {
      return null;
    } else {
      return /*#__PURE__*/jsxRuntime.jsx(TooltipContent.TooltipContent, {
        data: formattedData,
        theme: theme,
        title: title
      });
    }
  };
}

exports.getTooltipContentRenderer = getTooltipContentRenderer;
