// @shopify/polaris-viz/build/esm/components/SimpleNormalizedChart/SimpleNormalizedChart.js
import { usePolarisVizContext, DEFAULT_CHART_PROPS, ChartState } from '@shopify/polaris-viz-core';
import { Chart } from './Chart.js';
import { jsx } from 'react/jsx-runtime';
import { ChartContainer } from '../ChartContainer/ChartContainer.js';
import { ChartSkeleton } from '../ChartSkeleton/ChartSkeleton.js';

function SimpleNormalizedChart(props) {
  const {
    defaultTheme
  } = usePolarisVizContext();
  const {
    comparisonMetrics = [],
    data,
    labelFormatter = value => `${value}`,
    legendPosition = 'top-left',
    seriesNameFormatter = value => `${value}`,
    direction = 'horizontal',
    size = 'small',
    showLegend = true,
    theme = defaultTheme,
    id,
    isAnimated,
    state,
    errorText,
    renderLegendContent,
    onError
  } = { ...DEFAULT_CHART_PROPS,
    ...props
  };
  return /*#__PURE__*/jsx(ChartContainer, {
    data: data,
    theme: theme,
    id: id,
    isAnimated: isAnimated,
    onError: onError,
    children: state !== ChartState.Success ? /*#__PURE__*/jsx(ChartSkeleton, {
      type: "SimpleNormalized",
      state: state,
      errorText: errorText,
      theme: theme,
      showLegend: showLegend && !renderLegendContent,
      size: size
    }) : /*#__PURE__*/jsx(Chart, {
      comparisonMetrics: comparisonMetrics,
      data: data,
      labelFormatter: labelFormatter,
      legendPosition: legendPosition,
      seriesNameFormatter: seriesNameFormatter,
      showLegend: showLegend,
      direction: direction,
      size: size,
      renderLegendContent: renderLegendContent
    })
  });
}

export { SimpleNormalizedChart };
