// @shopify/polaris-viz/build/cjs/components/shared/ZeroValueLine/ZeroValueLine.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var constants = require('../../../constants.js');
var jsxRuntime = require('react/jsx-runtime');

function getZeroValueLineCoords({
  x,
  y,
  direction,
  areAllNegative
}) {
  if (direction === 'vertical') {
    return {
      x1: x,
      x2: x,
      y1: y,
      y2: !areAllNegative ? y - constants.ZERO_VALUE_LINE_HEIGHT : y + constants.ZERO_VALUE_LINE_HEIGHT
    };
  }

  return {
    x1: x,
    x2: !areAllNegative ? x + constants.ZERO_VALUE_LINE_HEIGHT : x - constants.ZERO_VALUE_LINE_HEIGHT,
    y1: y,
    y2: y
  };
}

function ZeroValueLine({
  x,
  y,
  theme,
  direction = 'vertical',
  areAllNegative
}) {
  const selectedTheme = polarisVizCore.useTheme(theme);
  return /*#__PURE__*/jsxRuntime.jsx(React.Fragment, {
    children: /*#__PURE__*/jsxRuntime.jsx("line", {
      stroke: selectedTheme.bar.zeroValueColor,
      strokeWidth: "1",
      ...getZeroValueLineCoords({
        x,
        y,
        direction,
        areAllNegative
      })
    })
  });
}

exports.ZeroValueLine = ZeroValueLine;
