// @shopify/polaris-viz-core/build/cjs/components/PolarisVizProvider/PolarisVizProvider.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('react');
var constants = require('../../constants.js');
var jsxRuntime = require('react/jsx-runtime');
var createThemes = require('../../utilities/createThemes.js');
var PolarisVizContext = require('../../contexts/PolarisVizContext.js');

function PolarisVizProvider({
  children,
  defaultTheme = constants.DEFAULT_THEME_NAME,
  themes,
  components,
  animated,
  onError
}) {
  const value = react.useMemo(() => {
    return {
      components: { ...constants.DEFAULT_COMPONENTS,
        ...components
      },
      themes: createThemes.createThemes({
        Dark: constants.DARK_THEME,
        Light: constants.LIGHT_THEME,
        Print: constants.PRINT_THEME,
        ...themes
      }),
      animated,
      defaultTheme,
      onError
    };
  }, [themes, components, animated, defaultTheme, onError]);
  return /*#__PURE__*/jsxRuntime.jsx(PolarisVizContext.PolarisVizContext.Provider, {
    value: value,
    children: children
  });
}

exports.PolarisVizProvider = PolarisVizProvider;
