// @shopify/polaris-viz/build/cjs/components/SparkBarChart/Chart.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var web = require('@react-spring/web');
var polarisVizCore = require('@shopify/polaris-viz-core');
var SparkBarChart = require('./SparkBarChart.scss.js');
var jsxRuntime = require('react/jsx-runtime');

function Chart({
  data,
  dimensions,
  targetLine,
  accessibilityLabel
}) {
  const {
    shouldAnimate
  } = polarisVizCore.useChartContext();
  const {
    width,
    height
  } = dimensions !== null && dimensions !== void 0 ? dimensions : {
    width: 0,
    height: 0
  };
  const viewboxHeight = height + polarisVizCore.ANIMATION_MARGIN * 2;
  return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
    children: [accessibilityLabel ? /*#__PURE__*/jsxRuntime.jsx("span", {
      className: SparkBarChart["default"].VisuallyHidden,
      children: accessibilityLabel
    }) : null, /*#__PURE__*/jsxRuntime.jsx("svg", {
      xmlns: polarisVizCore.XMLNS,
      "aria-hidden": true,
      viewBox: `0 -${polarisVizCore.ANIMATION_MARGIN} ${width} ${viewboxHeight}`,
      style: {
        transform: `translateY(-${polarisVizCore.ANIMATION_MARGIN}px)`
      },
      className: SparkBarChart["default"].Svg,
      height: viewboxHeight,
      width: width,
      children: /*#__PURE__*/jsxRuntime.jsx(polarisVizCore.SparkBarSeries, {
        data: data,
        targetLine: targetLine,
        height: height,
        shouldAnimate: shouldAnimate,
        useTransition: web.useTransition,
        width: width
      })
    })]
  });
}

exports.Chart = Chart;
