// @shopify/polaris-viz/build/cjs/components/SparkBarChart/SparkBarChart.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var Chart = require('./Chart.js');
var jsxRuntime = require('react/jsx-runtime');
var ChartContainer = require('../ChartContainer/ChartContainer.js');
var ChartSkeleton = require('../ChartSkeleton/ChartSkeleton.js');

function SparkBarChart(props) {
  const {
    defaultTheme
  } = polarisVizCore.usePolarisVizContext();
  const {
    data,
    accessibilityLabel,
    id,
    isAnimated,
    onError,
    targetLine,
    theme = defaultTheme,
    state,
    errorText
  } = { ...polarisVizCore.DEFAULT_CHART_PROPS,
    ...props
  };
  return /*#__PURE__*/jsxRuntime.jsx(ChartContainer.ChartContainer, {
    data: data,
    theme: theme,
    sparkChart: true,
    id: id,
    isAnimated: isAnimated,
    onError: onError,
    children: state !== polarisVizCore.ChartState.Success ? /*#__PURE__*/jsxRuntime.jsx(ChartSkeleton.ChartSkeleton, {
      type: "Spark",
      state: state,
      errorText: errorText,
      theme: theme
    }) : /*#__PURE__*/jsxRuntime.jsx(Chart.Chart, {
      data: data,
      targetLine: targetLine,
      accessibilityLabel: accessibilityLabel,
      theme: theme
    })
  });
}

exports.SparkBarChart = SparkBarChart;
