// @shopify/polaris-viz/build/cjs/components/ChartSkeleton/components/SimpleBarSkeleton/SimpleBarSkeleton.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var SimpleBarSkeleton$1 = require('./SimpleBarSkeleton.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var ErrorText = require('../ErrorText/ErrorText.js');

function SimpleBarSkeleton({
  dimensions,
  state,
  errorText
}) {
  const {
    width,
    height
  } = dimensions;
  const {
    grid: {
      color: gridColor
    },
    bar: {
      borderRadius
    }
  } = polarisVizCore.useTheme();
  const id = polarisVizCore.useUniqueId('simple-bar-skeleton');

  const BarMarkup = () => /*#__PURE__*/jsxRuntime.jsx("span", {
    style: {
      display: 'inline-block',
      background: gridColor,
      borderRadius
    }
  });

  return /*#__PURE__*/jsxRuntime.jsxs("div", {
    className: SimpleBarSkeleton$1["default"].SimpleBarSkeleton,
    children: [state === polarisVizCore.ChartState.Loading && /*#__PURE__*/jsxRuntime.jsx(React.Fragment, {
      children: new Array(4).fill(0).map((_, index) => /*#__PURE__*/jsxRuntime.jsxs("div", {
        className: SimpleBarSkeleton$1["default"].Item,
        children: [/*#__PURE__*/jsxRuntime.jsx(BarMarkup, {}), /*#__PURE__*/jsxRuntime.jsx(BarMarkup, {})]
      }, `${id}${index}`))
    }), state === polarisVizCore.ChartState.Error && /*#__PURE__*/jsxRuntime.jsx("svg", {
      viewBox: `0 0 ${width} ${height}`,
      children: /*#__PURE__*/jsxRuntime.jsx(ErrorText.ErrorText, {
        errorText: errorText,
        width: width,
        height: height
      })
    })]
  });
}

exports.SimpleBarSkeleton = SimpleBarSkeleton;
