// @shopify/polaris-viz/build/cjs/components/LineChart/hooks/useLineChartTooltipContent.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var React = require('react');

function useLineChartTooltipContent({
  data,
  hiddenIndexes = [],
  indexForLabels,
  renderTooltipContent,
  seriesNameFormatter
}) {
  const {
    theme
  } = polarisVizCore.useChartContext();
  return React.useCallback(activeIndex => {
    if (activeIndex === -1) {
      return null;
    }

    const tooltipData = [{
      shape: 'Line',
      data: []
    }];
    const key = data[indexForLabels].data[activeIndex] ? data[indexForLabels].data[activeIndex].key : '';
    data.forEach(({
      name,
      data: seriesData,
      color,
      isComparison
    }, index) => {
      if (!seriesData[activeIndex]) {
        return;
      }

      const {
        value
      } = seriesData[activeIndex];
      tooltipData[0].data.push({
        key: `${seriesNameFormatter(name !== null && name !== void 0 ? name : '')}`,
        value,
        color: color,
        isComparison,
        isHidden: hiddenIndexes.includes(index)
      });
    });
    return renderTooltipContent({
      data: tooltipData,
      activeIndex,
      title: `${key}`,
      dataSeries: data,
      theme
    });
  }, [data, renderTooltipContent, theme, hiddenIndexes, indexForLabels, seriesNameFormatter]);
}

exports.useLineChartTooltipContent = useLineChartTooltipContent;
