// @shopify/polaris-viz/build/esm/components/FunnelChart/components/FunnelChartXAxisLabels.js
import { Fragment } from 'react';
import { FunnelChartXAxisArrows } from './FunnelChartXAxisArrows.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { useLabels } from '../../Labels/hooks/useLabels.js';
import { shouldSkipLabel } from '../../Labels/utilities/shouldSkipLabel.js';
import { TextLine } from '../../TextLine/TextLine.js';

function FunnelChartXAxisLabels({
  allowLineWrap,
  chartX,
  chartY,
  labels,
  labelWidth,
  onHeightChange,
  reducedLabelIndexes,
  xScale
}) {
  const {
    lines
  } = useLabels({
    allowLineWrap,
    labels,
    targetWidth: labelWidth,
    onHeightChange
  });
  return /*#__PURE__*/jsx(Fragment, {
    children: lines.map((line, index) => {
      var _xScale;

      if (shouldSkipLabel(index, reducedLabelIndexes)) {
        return null;
      }

      const x = (_xScale = xScale(index.toString())) !== null && _xScale !== void 0 ? _xScale : 0;
      return /*#__PURE__*/jsxs("g", {
        children: [index === 0 ? null : /*#__PURE__*/jsx(FunnelChartXAxisArrows, {
          onHeightChange: onHeightChange,
          x: x,
          index: index,
          chartX: chartX,
          chartY: chartY,
          labelWidth: labelWidth
        }), /*#__PURE__*/jsx("g", {
          transform: `translate(${chartX + x},${chartY})`,
          children: /*#__PURE__*/jsx(TextLine, {
            line: line,
            index: index
          })
        }, index)]
      }, `label-group-${index}`);
    })
  });
}

export { FunnelChartXAxisLabels };
