// @shopify/polaris-viz/build/esm/components/FunnelChart/components/Label.js
import { animated } from '@react-spring/web';
import styles from './Label.scss.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { HORIZONTAL_BAR_LABEL_HEIGHT } from '@shopify/polaris-viz-core';

const LABEL_HEIGHT = 20;
const LARGE_FONT = 20;
const SMALL_FONT = 14;
const FONT_SIZES = {
  small: SMALL_FONT,
  large: LARGE_FONT
};
function Label({
  label,
  labelWidth,
  labelHelper,
  size,
  color,
  backgroundColor,
  transform
}) {
  const fontSize = FONT_SIZES[size];
  return (
    /*#__PURE__*/

    /* To display labels correctly in Safari, we need to wrap foreignObject in group element
    and apply the transform property on the group element. */
    jsx(animated.g, {
      "aria-hidden": true,
      style: {
        transform
      },
      children: /*#__PURE__*/jsx("foreignObject", {
        height: LABEL_HEIGHT,
        width: labelWidth,
        "aria-hidden": "true",
        className: styles.YAxis,
        children: /*#__PURE__*/jsxs("div", {
          className: styles.Label,
          style: {
            fontSize: `${fontSize}px`,
            color,
            lineHeight: `${HORIZONTAL_BAR_LABEL_HEIGHT}px`
          },
          children: [/*#__PURE__*/jsx("span", {
            style: {
              backgroundColor,
              height: HORIZONTAL_BAR_LABEL_HEIGHT,
              userSelect: 'none'
            },
            children: label
          }), labelHelper]
        })
      })
    })
  );
}

export { Label };
