// @shopify/polaris-viz/build/cjs/components/ComboChart/components/ComboBarChart/ComboBarChart.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var jsxRuntime = require('react/jsx-runtime');
var getAxisOptions = require('../../../../utilities/getAxisOptions.js');
var useVerticalBarChart = require('../../../VerticalBarChart/hooks/useVerticalBarChart.js');
var VerticalBarGroup = require('../../../VerticalBarChart/components/VerticalBarGroup/VerticalBarGroup.js');

function ComboBarChart({
  colors,
  data,
  drawableHeight,
  drawableWidth,
  indexOffset,
  labels,
  yScale
}) {
  const id = React.useMemo(() => polarisVizCore.uniqueId('ComboBarChart'), []);
  const yAxisOptionsWithDefaults = getAxisOptions.getYAxisOptionsWithDefaults(data.yAxisOptions);
  const {
    sortedData,
    xScale,
    gapWidth
  } = useVerticalBarChart.useVerticalBarChart({
    data: data.series,
    drawableWidth,
    labels
  });
  return /*#__PURE__*/jsxRuntime.jsx(VerticalBarGroup.VerticalBarGroup, {
    colors: colors,
    data: data.series,
    drawableHeight: drawableHeight,
    gapWidth: gapWidth,
    id: id,
    indexOffset: indexOffset,
    labels: labels,
    sortedData: sortedData,
    stackedValues: null,
    xScale: xScale,
    yAxisOptions: yAxisOptionsWithDefaults,
    yScale: yScale
  });
}

exports.ComboBarChart = ComboBarChart;
