// @shopify/polaris-viz/build/cjs/components/shared/GroupLabel/GroupLabel.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var GroupLabel$1 = require('./GroupLabel.scss.js');
var jsxRuntime = require('react/jsx-runtime');

const LABEL_RIGHT_PADDING = 15;
function GroupLabel({
  areAllNegative,
  containerWidth,
  label,
  zeroPosition
}) {
  const {
    characterWidths
  } = polarisVizCore.useChartContext();
  const labelWidth = polarisVizCore.estimateStringWidth(label, characterWidths);
  const selectedTheme = polarisVizCore.useTheme();
  const maxWidth = areAllNegative ? labelWidth : containerWidth - zeroPosition;
  return /*#__PURE__*/jsxRuntime.jsx("foreignObject", {
    height: polarisVizCore.HORIZONTAL_GROUP_LABEL_HEIGHT,
    width: "100%",
    x: zeroPosition + (areAllNegative ? labelWidth * -1 : 0),
    "aria-hidden": "true",
    children: /*#__PURE__*/jsxRuntime.jsx("div", {
      className: GroupLabel$1["default"].Label,
      style: {
        background: selectedTheme.chartContainer.backgroundColor,
        fontSize: `${polarisVizCore.FONT_SIZE}px`,
        color: selectedTheme.yAxis.labelColor,
        maxWidth,
        height: polarisVizCore.HORIZONTAL_GROUP_LABEL_HEIGHT,
        width: labelWidth + LABEL_RIGHT_PADDING
      },
      children: label
    })
  });
}

exports.GroupLabel = GroupLabel;
