// @shopify/polaris-viz/build/cjs/components/Annotations/Annotations.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var useSVGBlurEvent = require('../../hooks/useSVGBlurEvent.js');
var shouldHideAnnotation = require('../../utilities/shouldHideAnnotation.js');
var isShowMoreAnnotationsButtonVisible = require('../../utilities/isShowMoreAnnotationsButtonVisible.js');
var useAnnotationPositions = require('./hooks/useAnnotationPositions.js');
var constants = require('./constants.js');
var Annotations$1 = require('./Annotations.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var ShowMoreAnnotationsButton = require('./components/ShowMoreAnnotationsButton/ShowMoreAnnotationsButton.js');
var AnnotationLine = require('./components/AnnotationLine/AnnotationLine.js');
var AnnotationLabel = require('./components/AnnotationLabel/AnnotationLabel.js');
var AnnotationContent = require('./components/AnnotationContent/AnnotationContent.js');

function Annotations({
  annotationsLookupTable,
  axisLabelWidth,
  drawableHeight,
  drawableWidth,
  labels,
  onHeightChange,
  xScale
}) {
  const [activeIndex, setActiveIndex] = React.useState(-1);
  const [isShowingAllAnnotations, setIsShowingAllAnnotations] = React.useState(false);
  const [ref, setRef] = React.useState(null);
  const {
    annotations,
    dataIndexes
  } = React.useMemo(() => {
    const dataIndexes = {};
    const annotations = Object.keys(annotationsLookupTable).map(key => {
      const annotation = annotationsLookupTable[key];

      if (!labels.includes(key) || annotation == null || annotation.axis === 'y') {
        return null;
      }

      dataIndexes[key] = labels.indexOf(key);
      return annotation;
    }).filter(Boolean);
    return {
      annotations,
      dataIndexes
    };
  }, [annotationsLookupTable, labels]);
  const {
    hiddenAnnotationsCount,
    positions,
    rowCount
  } = useAnnotationPositions.useAnnotationPositions({
    annotations,
    axisLabelWidth,
    dataIndexes,
    drawableWidth,
    isShowingAllAnnotations,
    onHeightChange,
    xScale
  });

  const handleToggleAllAnnotations = () => {
    setIsShowingAllAnnotations(!isShowingAllAnnotations);
  };

  const handleOnMouseLeave = () => {
    setActiveIndex(-1);
  };

  useSVGBlurEvent.useSVGBlurEvent({
    ref,
    onBlur: handleOnMouseLeave,
    checkFn: activeElement => {
      const focusedParent = activeElement === null || activeElement === void 0 ? void 0 : activeElement.parentElement;
      return (focusedParent === null || focusedParent === void 0 ? void 0 : focusedParent.dataset.isAnnotationContent) !== 'true';
    }
  });
  const isShowMoreButtonVisible = isShowMoreAnnotationsButtonVisible.isShowMoreAnnotationsButtonVisible(rowCount);
  const showMoreButtonOffset = isShowMoreButtonVisible ? constants.SHOW_MORE_BUTTON_OFFSET : 0;
  return /*#__PURE__*/jsxRuntime.jsxs("g", {
    ref: setRef,
    tabIndex: -1,
    className: Annotations$1["default"].Group,
    children: [isShowMoreButtonVisible && /*#__PURE__*/jsxRuntime.jsx(ShowMoreAnnotationsButton.ShowMoreAnnotationsButton, {
      annotationsCount: hiddenAnnotationsCount,
      collapseText: annotations[0].collapseButtonText,
      expandText: annotations[0].expandButtonText,
      isShowingAllAnnotations: isShowingAllAnnotations,
      onClick: handleToggleAllAnnotations,
      tabIndex: annotations.length,
      width: drawableWidth
    }), /*#__PURE__*/jsxRuntime.jsx("g", {
      transform: `translate(0, ${showMoreButtonOffset})`,
      children: positions.map(position => {
        const {
          line,
          y,
          row,
          index
        } = position;
        const annotation = annotations[index];

        if (shouldHideAnnotation.shouldHideAnnotation({
          row,
          isShowingAllAnnotations,
          rowCount
        })) {
          return null;
        }

        const hasContent = annotation.content != null;
        const isContentVisible = index === activeIndex && hasContent;
        const tabIndex = index + 1;
        const ariaLabel = `${annotation.startKey}`;
        return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
          children: [/*#__PURE__*/jsxRuntime.jsx(AnnotationLine.AnnotationLine, {
            size: drawableHeight - showMoreButtonOffset,
            x: line.x,
            y: y + constants.PILL_HEIGHT
          }), /*#__PURE__*/jsxRuntime.jsx(AnnotationLabel.AnnotationLabel, {
            ariaLabel: ariaLabel,
            hasContent: hasContent,
            index: index,
            isVisible: !isContentVisible,
            label: annotation.label,
            position: position,
            setActiveIndex: setActiveIndex,
            tabIndex: tabIndex
          }), isContentVisible && /*#__PURE__*/jsxRuntime.jsx(AnnotationContent.AnnotationContent, {
            annotation: annotation,
            drawableWidth: drawableWidth,
            index: index,
            onMouseLeave: handleOnMouseLeave,
            parentRef: ref,
            position: position,
            tabIndex: tabIndex,
            x: line.x,
            y: y
          })]
        }, `annotation${index}${annotation.startKey}`);
      })
    })]
  });
}

exports.Annotations = Annotations;
