// @shopify/polaris-viz/build/cjs/components/Annotations/YAxisAnnotations.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var useSVGBlurEvent = require('../../hooks/useSVGBlurEvent.js');
var useYAxisAnnotationPositions = require('./hooks/useYAxisAnnotationPositions.js');
var Annotations = require('./Annotations.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var AnnotationYAxisLabel = require('./components/AnnotationYAxisLabel/AnnotationYAxisLabel.js');
var AnnotationLine = require('./components/AnnotationLine/AnnotationLine.js');
var AnnotationLabel = require('./components/AnnotationLabel/AnnotationLabel.js');
var AnnotationContent = require('./components/AnnotationContent/AnnotationContent.js');

function YAxisAnnotations({
  axis = 'y',
  annotationsLookupTable,
  drawableWidth,
  ticks,
  yScale
}) {
  const [activeIndex, setActiveIndex] = React.useState(-1);
  const [ref, setRef] = React.useState(null);
  const {
    annotations
  } = React.useMemo(() => {
    const annotations = Object.keys(annotationsLookupTable).map(key => {
      const annotation = annotationsLookupTable[key];

      if (!polarisVizCore.isValueWithinDomain(Number(annotation.startKey), yScale.domain())) {
        return null;
      }

      if (annotation == null || annotation.axis == null || annotation.axis !== axis) {
        return null;
      }

      return annotation;
    }).filter(Boolean);
    return {
      annotations
    };
  }, [annotationsLookupTable, yScale, axis]);
  const {
    positions
  } = useYAxisAnnotationPositions.useYAxisAnnotationPositions({
    axis,
    annotations,
    drawableWidth,
    ticks,
    yScale
  });

  const handleOnMouseLeave = () => {
    setActiveIndex(-1);
  };

  useSVGBlurEvent.useSVGBlurEvent({
    ref,
    onBlur: handleOnMouseLeave,
    checkFn: activeElement => {
      const focusedParent = activeElement === null || activeElement === void 0 ? void 0 : activeElement.parentElement;
      return (focusedParent === null || focusedParent === void 0 ? void 0 : focusedParent.dataset.isAnnotationContent) !== 'true';
    }
  });
  return /*#__PURE__*/jsxRuntime.jsx("g", {
    ref: setRef,
    tabIndex: -1,
    className: Annotations["default"].Group,
    children: positions.map(position => {
      var _line$width;

      const index = position.index;
      const annotation = annotations[index];
      const {
        line,
        x,
        y
      } = position;
      const hasContent = annotation.content != null;
      const isContentVisible = index === activeIndex && hasContent;
      const tabIndex = index + 1;
      const ariaLabel = `${annotation.startKey}`;
      const axisLabelX = axis === 'y2' ? drawableWidth + polarisVizCore.Y_AXIS_CHART_SPACING : -polarisVizCore.Y_AXIS_CHART_SPACING;
      return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
        children: [position.showYAxisLabel && /*#__PURE__*/jsxRuntime.jsx(AnnotationYAxisLabel.AnnotationYAxisLabel, {
          axis: axis,
          y: line.y,
          x: axisLabelX,
          label: annotation.startKey
        }), /*#__PURE__*/jsxRuntime.jsx(AnnotationLine.AnnotationLine, {
          direction: "horizontal",
          hasCaret: false,
          size: (_line$width = line.width) !== null && _line$width !== void 0 ? _line$width : 0,
          x: line.x,
          y: line.y
        }), /*#__PURE__*/jsxRuntime.jsx(AnnotationLabel.AnnotationLabel, {
          ariaLabel: ariaLabel,
          hasContent: hasContent,
          index: index,
          isVisible: !isContentVisible,
          label: annotation.label,
          position: position,
          setActiveIndex: setActiveIndex,
          tabIndex: tabIndex
        }), isContentVisible && /*#__PURE__*/jsxRuntime.jsx(AnnotationContent.AnnotationContent, {
          annotation: annotation,
          drawableWidth: drawableWidth,
          index: index,
          onMouseLeave: handleOnMouseLeave,
          parentRef: ref,
          position: position,
          tabIndex: tabIndex,
          x: drawableWidth - (drawableWidth - x),
          y: y
        })]
      }, `annotation${index}${annotation.startKey}`);
    })
  });
}

exports.YAxisAnnotations = YAxisAnnotations;
