// @shopify/polaris-viz/build/esm/components/DonutChart/components/LegendValues/LegendValues.js
import { useTheme, useChartContext, clamp, getSeriesColors, estimateStringWidth, COLOR_VISION_SINGLE_ITEM } from '@shopify/polaris-viz-core';
import React__default, { useRef, useMemo } from 'react';
import { useOverflowLegend } from '../../../LegendContainer/hooks/useOverflowLegend.js';
import { getTrendIndicatorData } from '../../../../utilities/getTrendIndicatorData.js';
import { HiddenLegendTooltip } from '../../../LegendContainer/components/HiddenLegendTooltip.js';
import { useLegend } from '../../../LegendContainer/hooks/useLegend.js';
import styles from './LegendValues.scss.js';
import { LegendValueItem } from './components/LegendValueItem/LegendValueItem.js';
import { jsxs, jsx } from 'react/jsx-runtime';

/* eslint-disable @shopify/strict-component-boundaries */
const TABLE_LEGEND_PADDING = 50;
function LegendValues({
  data: allData,
  activeIndex,
  dimensions,
  legendFullWidth,
  labelFormatter,
  renderHiddenLegendLabel = count => `+${count} more`,
  getColorVisionStyles,
  getColorVisionEventAttrs,
  seriesNameFormatter
}) {
  const selectedTheme = useTheme();
  const {
    theme,
    characterWidths
  } = useChartContext();
  const legendItemDimensions = useRef([{
    width: 0,
    height: 0
  }]);
  const seriesCount = clamp({
    amount: allData.length,
    min: 1,
    max: Infinity
  });
  const seriesColors = getSeriesColors(seriesCount, selectedTheme);
  const {
    legend: legendData,
    height
  } = useLegend({
    showLegend: true,
    data: [{
      series: allData,
      shape: 'Bar'
    }],
    colors: seriesColors,
    dimensions,
    seriesNameFormatter
  });
  const {
    displayedData,
    hiddenData
  } = useOverflowLegend({
    direction: 'vertical',
    data: legendData,
    height,
    legendItemDimensions,
    enableHideOverflow: true
  });
  const hasHiddenData = displayedData.length < allData.length;
  const maxTrendIndicatorWidth = allData.reduce((maxWidth, {
    metadata
  }) => {
    if (!(metadata !== null && metadata !== void 0 && metadata.trend)) {
      return maxWidth;
    }

    const {
      trendIndicatorWidth
    } = getTrendIndicatorData(metadata.trend);
    return Math.max(maxWidth, trendIndicatorWidth);
  }, 0);
  const longestLegendNameWidth = useMemo(() => {
    return legendData.reduce((previous, current) => {
      const estimatedLegendNameWidth = estimateStringWidth(`${seriesNameFormatter(`${current.name || ''}`)}`, characterWidths);

      if (estimatedLegendNameWidth > previous) {
        return estimatedLegendNameWidth;
      }

      return previous;
    }, 0);
  }, [legendData, seriesNameFormatter, characterWidths]);
  const longestLegendValueWidth = useMemo(() => {
    return legendData.reduce((previous, current) => {
      const estimatedLegendValueWidth = estimateStringWidth(`${labelFormatter(`${current.value || ''}`)}`, characterWidths);

      if (estimatedLegendValueWidth > previous) {
        return estimatedLegendValueWidth;
      }

      return previous;
    }, 0);
  }, [legendData, labelFormatter, characterWidths]);
  const legendTableMaxWidth = useMemo(() => longestLegendNameWidth + longestLegendValueWidth + maxTrendIndicatorWidth + TABLE_LEGEND_PADDING, [longestLegendNameWidth, longestLegendValueWidth, maxTrendIndicatorWidth]);
  return /*#__PURE__*/jsxs(React__default.Fragment, {
    children: [/*#__PURE__*/jsx("table", {
      className: styles.Table,
      style: {
        maxWidth: legendTableMaxWidth,
        width: legendFullWidth ? 'auto' : '100%'
      },
      children: /*#__PURE__*/jsx("tbody", {
        children: displayedData.map(({
          name,
          value,
          trend
        }, index) => {
          return /*#__PURE__*/jsx(LegendValueItem, {
            name: name,
            value: value,
            trend: trend,
            index: index,
            longestLegendValueWidth: longestLegendValueWidth,
            maxTrendIndicatorWidth: maxTrendIndicatorWidth,
            seriesColors: seriesColors,
            seriesNameFormatter: seriesNameFormatter,
            onDimensionChange: dimensions => {
              if (legendItemDimensions.current) {
                legendItemDimensions.current[index] = dimensions;
              }
            },
            getColorVisionEventAttrs: getColorVisionEventAttrs,
            getColorVisionStyles: getColorVisionStyles,
            labelFormatter: labelFormatter
          }, index);
        })
      })
    }), hasHiddenData && /*#__PURE__*/jsx(HiddenLegendTooltip, {
      activeIndex: activeIndex,
      colorVisionType: COLOR_VISION_SINGLE_ITEM,
      data: hiddenData,
      theme: theme,
      label: renderHiddenLegendLabel(allData.length - displayedData.length),
      lastVisibleIndex: allData.length - hiddenData.length,
      setActivatorWidth: () => null,
      dimensions: dimensions,
      seriesNameFormatter: seriesNameFormatter
    })]
  });
}

export { LegendValues };
