// @shopify/polaris-viz/build/esm/components/FunnelChart/FunnelChart.js
import { usePolarisVizContext, DEFAULT_CHART_PROPS, useTheme, useThemeSeriesColors, ChartState } from '@shopify/polaris-viz-core';
import { Chart } from './Chart.js';
import { jsx } from 'react/jsx-runtime';
import { getXAxisOptionsWithDefaults, getYAxisOptionsWithDefaults } from '../../utilities/getAxisOptions.js';
import { ChartContainer } from '../ChartContainer/ChartContainer.js';
import { ChartSkeleton } from '../ChartSkeleton/ChartSkeleton.js';

function FunnelChart(props) {
  const {
    defaultTheme
  } = usePolarisVizContext();
  const {
    data,
    theme = defaultTheme,
    xAxisOptions,
    yAxisOptions,
    id,
    isAnimated,
    state,
    errorText,
    labelHelpers,
    onError
  } = { ...DEFAULT_CHART_PROPS,
    ...props
  };
  const selectedTheme = useTheme(theme);
  const seriesColors = useThemeSeriesColors(data, selectedTheme);
  const seriesWithDefaults = data.map((series, index) => ({
    color: seriesColors[index],
    ...series
  }));
  const xAxisOptionsForChart = getXAxisOptionsWithDefaults(xAxisOptions);
  const yAxisOptionsForChart = getYAxisOptionsWithDefaults(yAxisOptions);
  return /*#__PURE__*/jsx(ChartContainer, {
    data: data,
    id: id,
    isAnimated: isAnimated,
    onError: onError,
    theme: theme,
    children: state !== ChartState.Success ? /*#__PURE__*/jsx(ChartSkeleton, {
      type: "Funnel",
      state: state,
      errorText: errorText,
      theme: theme
    }) : /*#__PURE__*/jsx(Chart, {
      data: seriesWithDefaults,
      labelHelpers: labelHelpers,
      xAxisOptions: xAxisOptionsForChart,
      yAxisOptions: yAxisOptionsForChart
    })
  });
}

export { FunnelChart };
