// @shopify/polaris-viz/build/cjs/components/LineChartRelational/components/Area/Area.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var web = require('@react-spring/web');
var polarisVizCore = require('@shopify/polaris-viz-core');
var jsxRuntime = require('react/jsx-runtime');

function Area({
  activeIndex,
  fill,
  getAreaGenerator,
  hiddenIndexes = [],
  index,
  series,
  shouldAnimate = false
}) {
  var _series$metadata$rela, _series$metadata;

  const pathD = getAreaGenerator(series);
  const previous = polarisVizCore.usePrevious(pathD);
  const spring = web.useSpring({
    from: {
      pathD: previous
    },
    to: {
      pathD
    },
    immediate: !shouldAnimate
  });

  if (hiddenIndexes.includes(index)) {
    return null;
  }

  if (hiddenIndexes.includes((_series$metadata$rela = (_series$metadata = series.metadata) === null || _series$metadata === void 0 ? void 0 : _series$metadata.relatedIndex) !== null && _series$metadata$rela !== void 0 ? _series$metadata$rela : -1)) {
    return null;
  }

  if (pathD == null) {
    return null;
  }

  return /*#__PURE__*/jsxRuntime.jsx(web.animated.path, {
    d: spring.pathD,
    fill: fill,
    style: { ...polarisVizCore.getColorVisionStylesForActiveIndex({
        activeIndex,
        index,
        fadedOpacity: 0.2
      })
    }
  });
}

exports.Area = Area;
