// @shopify/polaris-viz/build/cjs/components/ChartErrorBoundary/ChartErrorBoundary.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var React = require('react');
var checkForMismatchedData = require('./utilities/checkForMismatchedData.js');
var jsxRuntime = require('react/jsx-runtime');
var ChartSkeleton = require('../ChartSkeleton/ChartSkeleton.js');

class ChartErrorBoundary extends React.Component {
  static getDerivedStateFromError(_) {
    return {
      hasError: true
    };
  }

  constructor(props) {
    super(props);
    this.state = {
      hasError: false
    };

    this.onError = (error, errorInfo) => {
      checkForMismatchedData.checkForMismatchedData(this.props.data);

      if (this.props.onError) {
        this.props.onError(error, errorInfo);
      } else {
        // eslint-disable-next-line no-console
        console.error({
          error,
          errorInfo
        });
      }
    };
  }

  componentDidCatch(error, errorInfo) {
    this.onError(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      return /*#__PURE__*/jsxRuntime.jsx(ChartSkeleton.ChartSkeleton, {
        type: this.props.type,
        state: polarisVizCore.ChartState.Error,
        dimensions: this.props.dimensions
      });
    }

    return this.props.children;
  }

}

exports.ChartErrorBoundary = ChartErrorBoundary;
