// @shopify/polaris-viz/build/cjs/components/Labels/SingleTextLine.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var endLineTruncate = require('./utilities/endLineTruncate.js');
var jsxRuntime = require('react/jsx-runtime');

function SingleTextLine({
  ariaHidden = false,
  color,
  dominantBaseline = 'hanging',
  targetWidth,
  text,
  textAnchor = 'center',
  y,
  x
}) {
  const {
    characterWidths
  } = polarisVizCore.useChartContext();
  const truncated = endLineTruncate.endLineTruncate({
    label: text,
    targetWidth,
    targetHeight: polarisVizCore.LINE_HEIGHT,
    characterWidths
  });
  return /*#__PURE__*/jsxRuntime.jsxs(React.Fragment, {
    children: [/*#__PURE__*/jsxRuntime.jsx("text", {
      "aria-hidden": ariaHidden,
      textAnchor: textAnchor,
      dominantBaseline: dominantBaseline,
      height: polarisVizCore.LINE_HEIGHT,
      width: targetWidth,
      fill: color,
      fontSize: polarisVizCore.FONT_SIZE,
      fontFamily: polarisVizCore.FONT_FAMILY,
      y: y,
      x: x,
      children: truncated
    }), /*#__PURE__*/jsxRuntime.jsx("title", {
      children: text
    })]
  });
}

exports.SingleTextLine = SingleTextLine;
