// @shopify/polaris-viz/build/esm/components/HorizontalBarChart/components/HorizontalBarChartXAnnotations/hooks/useHorizontalBarChartXAnnotationPositions.js
import { useMemo, useEffect } from 'react';
import { useChartContext, estimateStringWidth, clamp } from '@shopify/polaris-viz-core';
import { COLLAPSED_ANNOTATIONS_COUNT } from '../../../../../constants.js';
import { PILL_PADDING, PILL_X_MIN, PILL_HEIGHT, PILL_ROW_GAP } from '../../../../Annotations/constants.js';
import { checkForHorizontalSpace } from '../../../../Annotations/utilities/checkForHorizontalSpace.js';
import { useShowMoreAnnotationsButton } from '../../../../Annotations/hooks/useShowMoreAnnotationsButton.js';

function useHorizontalBarChartXAnnotationPositions({
  annotations,
  drawableWidth,
  isShowingAllAnnotations,
  onHeightChange,
  xScale
}) {
  const {
    characterWidths
  } = useChartContext();
  const textWidths = useMemo(() => {
    return annotations.map(annotation => {
      return estimateStringWidth(annotation.label, characterWidths);
    });
  }, [annotations, characterWidths]);
  const {
    positions,
    hiddenAnnotationsCount
  } = useMemo(() => {
    let positions = annotations.map((annotation, dataIndex) => {
      const xPosition = xScale(Number(annotation.startKey));
      const textWidth = textWidths[dataIndex];
      const width = clamp({
        amount: textWidth + PILL_PADDING * 2,
        min: textWidth + PILL_PADDING * 2,
        max: drawableWidth
      });
      let x = clamp({
        amount: xPosition - width / 2,
        min: PILL_X_MIN,
        max: Infinity
      });
      const right = x + width;

      if (right > drawableWidth) {
        x -= right - drawableWidth;
      }

      return {
        index: dataIndex,
        line: {
          x: xPosition,
          y: 0
        },
        row: 1,
        width,
        x,
        y: 0
      };
    });
    positions = positions.sort((one, two) => one.x - two.x);
    checkForHorizontalSpace({
      positions,
      totalRows: 1
    });
    positions.forEach(current => {
      const row = current.row - 1;
      current.y = row * PILL_HEIGHT + row * PILL_ROW_GAP;
    });
    const hiddenAnnotationsCount = positions.filter(({
      row
    }) => row >= COLLAPSED_ANNOTATIONS_COUNT).length;
    return {
      positions,
      hiddenAnnotationsCount
    };
  }, [annotations, textWidths, xScale, drawableWidth]);
  const {
    rowCount,
    totalRowHeight
  } = useShowMoreAnnotationsButton({
    isShowingAllAnnotations,
    positions
  });
  useEffect(() => {
    onHeightChange(totalRowHeight);
  }, [onHeightChange, totalRowHeight]);
  return {
    positions,
    rowCount,
    hiddenAnnotationsCount: isShowingAllAnnotations ? 0 : hiddenAnnotationsCount
  };
}

export { useHorizontalBarChartXAnnotationPositions };
