// @shopify/polaris-viz/build/esm/components/shared/HorizontalStackedBars/HorizontalStackedBars.js
import { useState, useMemo, Fragment } from 'react';
import { useTheme, useChartContext, COLOR_VISION_SINGLE_ITEM, estimateStringWidth, HORIZONTAL_BAR_LABEL_OFFSET, HORIZONTAL_GROUP_LABEL_HEIGHT, BORDER_RADIUS } from '@shopify/polaris-viz-core';
import { NEGATIVE_ZERO_LINE_OFFSET } from '../../../constants.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { useWatchColorVisionEvents } from '../../../hooks/ColorVisionA11y/useWatchColorVisionEvents.js';
import { useStackedGaps } from './hooks/useStackedGaps.js';
import { getXPosition } from './utilities/getXPosition.js';
import { ZeroValueLine } from '../ZeroValueLine/ZeroValueLine.js';
import { StackedBar } from './components/StackedBar/StackedBar.js';
import { getGradientDefId } from '../GradientDefs/GradientDefs.js';
import { LabelWrapper } from '../LabelWrapper/LabelWrapper.js';
import { Label } from '../Label/Label.js';
import { getBarId } from '../../../utilities/getBarId.js';

function getBorderRadius({
  lastIndexes,
  seriesIndex
}) {
  const [positive, negative] = lastIndexes;

  if (positive === seriesIndex) {
    return BORDER_RADIUS.Right;
  }

  if (negative === seriesIndex) {
    return BORDER_RADIUS.Left;
  }

  return BORDER_RADIUS.None;
}

function HorizontalStackedBars({
  activeGroupIndex,
  animationDelay,
  barHeight,
  dataKeys,
  groupIndex,
  id,
  name,
  stackedValues,
  xScale,
  areAllNegative,
  isSimple,
  labelFormatter
}) {
  const selectedTheme = useTheme();
  const {
    theme,
    characterWidths
  } = useChartContext();
  const [activeBarIndex, setActiveBarIndex] = useState(-1);
  const hideGroupLabel = selectedTheme.groupLabel.hide;
  useWatchColorVisionEvents({
    type: COLOR_VISION_SINGLE_ITEM,
    onIndexChange: ({
      detail
    }) => {
      if (activeGroupIndex === -1 || activeGroupIndex === groupIndex) {
        setActiveBarIndex(detail.index);
      }
    }
  });
  const lastIndexes = useMemo(() => {
    let lastPos = -1;
    let lastNeg = -1;
    stackedValues[groupIndex].forEach(([start, end], index) => {
      if (start < 0) {
        lastNeg = index;
      }

      if (end > 0) {
        lastPos = index;
      }
    });
    return [lastPos, lastNeg];
  }, [groupIndex, stackedValues]);
  const gaps = useStackedGaps({
    stackedValues,
    groupIndex
  });
  const groupSum = stackedValues[groupIndex].reduce((_, item) => {
    if (item.data) {
      return Object.values(item.data).reduce((sum, value) => sum + value, 0);
    }

    return 0;
  }, 0);
  const isNegative = groupSum && groupSum < 0;
  const label = labelFormatter(groupSum);
  const labelWidth = estimateStringWidth(`${label}`, characterWidths);
  const minGroupStartPoint = stackedValues[groupIndex].reduce((min, item) => {
    const start = item[0];
    return start < min ? start : min;
  }, Infinity);
  const maxGroupEndPoint = stackedValues[groupIndex].reduce((max, item) => {
    const end = item[1];
    return end > max ? end : max;
  }, -Infinity);
  const groupLabelX = isNegative ? xScale(minGroupStartPoint) - labelWidth - HORIZONTAL_BAR_LABEL_OFFSET : xScale(maxGroupEndPoint) + HORIZONTAL_BAR_LABEL_OFFSET;
  return /*#__PURE__*/jsxs("g", {
    style: {
      transform: `translate(0, ${HORIZONTAL_GROUP_LABEL_HEIGHT}px`
    },
    "aria-hidden": "true",
    children: [stackedValues[groupIndex].map((item, seriesIndex) => {
      var _dataKeys$seriesIndex;

      const [start, end] = item;
      const barId = getBarId(id, groupIndex, seriesIndex);
      const width = Math.abs(xScale(end) - xScale(start));
      const borderRadius = getBorderRadius({
        lastIndexes,
        seriesIndex
      });
      const x = getXPosition({
        start,
        end,
        seriesIndex,
        gaps,
        xScale
      });
      const key = (_dataKeys$seriesIndex = dataKeys[seriesIndex]) !== null && _dataKeys$seriesIndex !== void 0 ? _dataKeys$seriesIndex : '';
      const ariaLabel = `${key} ${end}`;
      const areAllValuesZero = stackedValues[groupIndex].every(([start, end]) => start + end === 0);
      return /*#__PURE__*/jsx(Fragment, {
        children: areAllValuesZero ? /*#__PURE__*/jsx(ZeroValueLine, {
          x: areAllNegative ? x - NEGATIVE_ZERO_LINE_OFFSET : x,
          y: barHeight / 2,
          direction: "horizontal"
        }) : /*#__PURE__*/jsx(StackedBar, {
          animationDelay: animationDelay,
          activeBarIndex: activeBarIndex,
          ariaLabel: ariaLabel,
          borderRadius: borderRadius,
          color: getGradientDefId(theme, seriesIndex, id),
          height: barHeight,
          seriesIndex: seriesIndex,
          setActiveBarIndex: setActiveBarIndex,
          width: width,
          x: x,
          zeroPosition: xScale(0)
        }, `${name}${barId}`)
      }, `stackedBar ${barId}`);
    }), !isSimple && !hideGroupLabel && /*#__PURE__*/jsx(LabelWrapper, {
      animationDelay: animationDelay,
      x: groupLabelX,
      children: /*#__PURE__*/jsx(Label, {
        barHeight: barHeight,
        color: selectedTheme.xAxis.labelColor,
        label: label,
        labelWidth: labelWidth,
        y: 0
      })
    })]
  });
}

export { HorizontalStackedBars };
