// @shopify/polaris-viz/build/cjs/components/Labels/utilities/getVerticalLabels.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var truncateLabels = require('./truncateLabels.js');

const QUARTER = 4;
function getVerticalLabels({
  labels,
  characterWidths,
  longestLabelWidth,
  targetWidth
}) {
  const clampedTargetWidth = polarisVizCore.clamp({
    amount: longestLabelWidth,
    min: targetWidth,
    max: polarisVizCore.VERTICAL_LABEL_TARGET_WIDTH
  });
  const truncatedLabels = truncateLabels.truncateLabels({
    labels,
    targetWidth: clampedTargetWidth,
    targetHeight: polarisVizCore.LINE_HEIGHT,
    characterWidths,
    skipEndWordTruncate: true
  });
  const lines = [];
  let longestString = 0;

  for (let i = 0; i < labels.length; i++) {
    lines[i] = [];
    lines[i].push({
      truncatedText: truncatedLabels[i].truncatedName,
      fullText: truncatedLabels[i].text,
      y: polarisVizCore.LINE_HEIGHT / QUARTER,
      x: 0,
      width: clampedTargetWidth,
      height: polarisVizCore.LINE_HEIGHT,
      textAnchor: 'end',
      transform: `translate(${targetWidth / 2}) rotate(-90)`
    });

    if (truncatedLabels[i].truncatedWidth > longestString) {
      longestString = truncatedLabels[i].truncatedWidth;
    }
  }

  return {
    lines,
    containerHeight: longestString
  };
}

exports.getVerticalLabels = getVerticalLabels;
