// @shopify/polaris-viz/build/cjs/components/ChartSkeleton/components/ErrorText/ErrorText.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var jsxRuntime = require('react/jsx-runtime');
var useLabels = require('../../../Labels/hooks/useLabels.js');
var TextLine = require('../../../TextLine/TextLine.js');

const TEXT_DROP_SHADOW_SIZE = 3;
function ErrorText({
  errorText,
  width,
  height
}) {
  const {
    chartContainer: {
      backgroundColor
    }
  } = polarisVizCore.useTheme();
  const {
    lines
  } = useLabels.useLabels({
    allowLineWrap: true,
    labels: [errorText],
    targetWidth: width
  });
  return /*#__PURE__*/jsxRuntime.jsx("g", {
    style: {
      transform: `translateY(${Math.max(0, height / 2 - polarisVizCore.FONT_SIZE / 2)}px)`,
      filter: `
        drop-shadow( ${TEXT_DROP_SHADOW_SIZE}px 0px 1px ${backgroundColor})
        drop-shadow( -${TEXT_DROP_SHADOW_SIZE}px  0px 1px ${backgroundColor})
        drop-shadow( 0px ${TEXT_DROP_SHADOW_SIZE}px 1px ${backgroundColor})
        drop-shadow( 0px -${TEXT_DROP_SHADOW_SIZE}px 1px ${backgroundColor})
      `
    },
    children: /*#__PURE__*/jsxRuntime.jsx(TextLine.TextLine, {
      index: 0,
      line: lines[0]
    })
  });
}

exports.ErrorText = ErrorText;
