// @shopify/polaris-viz/build/cjs/components/FunnelChart/FunnelChart.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var Chart = require('./Chart.js');
var jsxRuntime = require('react/jsx-runtime');
var getAxisOptions = require('../../utilities/getAxisOptions.js');
var ChartContainer = require('../ChartContainer/ChartContainer.js');
var ChartSkeleton = require('../ChartSkeleton/ChartSkeleton.js');

function FunnelChart(props) {
  const {
    defaultTheme
  } = polarisVizCore.usePolarisVizContext();
  const {
    data,
    theme = defaultTheme,
    xAxisOptions,
    yAxisOptions,
    id,
    isAnimated,
    state,
    errorText,
    labelHelpers,
    onError
  } = { ...polarisVizCore.DEFAULT_CHART_PROPS,
    ...props
  };
  const selectedTheme = polarisVizCore.useTheme(theme);
  const seriesColors = polarisVizCore.useThemeSeriesColors(data, selectedTheme);
  const seriesWithDefaults = data.map((series, index) => ({
    color: seriesColors[index],
    ...series
  }));
  const xAxisOptionsForChart = getAxisOptions.getXAxisOptionsWithDefaults(xAxisOptions);
  const yAxisOptionsForChart = getAxisOptions.getYAxisOptionsWithDefaults(yAxisOptions);
  return /*#__PURE__*/jsxRuntime.jsx(ChartContainer.ChartContainer, {
    data: data,
    id: id,
    isAnimated: isAnimated,
    onError: onError,
    theme: theme,
    children: state !== polarisVizCore.ChartState.Success ? /*#__PURE__*/jsxRuntime.jsx(ChartSkeleton.ChartSkeleton, {
      type: "Funnel",
      state: state,
      errorText: errorText,
      theme: theme
    }) : /*#__PURE__*/jsxRuntime.jsx(Chart.Chart, {
      data: seriesWithDefaults,
      labelHelpers: labelHelpers,
      xAxisOptions: xAxisOptionsForChart,
      yAxisOptions: yAxisOptionsForChart
    })
  });
}

exports.FunnelChart = FunnelChart;
