// @shopify/polaris-viz/build/esm/components/VerticalBarChart/Chart.js
import { useState, useMemo } from 'react';
import { useTheme, useChartContext, uniqueId, LINE_HEIGHT, useYScale, estimateStringWidth, useChartPositions, DataType, ChartMargin, COLOR_VISION_SINGLE_ITEM } from '@shopify/polaris-viz-core';
import { stackOrderNone, stackOffsetDiverging } from 'd3-shape';
import { useIndexForLabels } from '../../hooks/useIndexForLabels.js';
import { ChartElements } from '../ChartElements/index.js';
import { useFormattedLabels } from '../../hooks/useFormattedLabels.js';
import { getVerticalBarChartTooltipPosition } from '../../utilities/getVerticalBarChartTooltipPosition.js';
import { ANNOTATIONS_LABELS_OFFSET } from '../../constants.js';
import { useVerticalBarChart } from './hooks/useVerticalBarChart.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { useColorVisionEvents } from '../../hooks/ColorVisionA11y/useColorVisionEvents.js';
import { useLegend } from '../LegendContainer/hooks/useLegend.js';
import { getStackedValues } from '../../utilities/getStackedValues.js';
import { useReducedLabelIndexes } from '../../hooks/useReducedLabelIndexes.js';
import { getStackedMinMax } from '../../utilities/getStackedMinMax.js';
import { useBarChartTooltipContent } from '../../hooks/useBarChartTooltipContent.js';
import { checkAvailableAnnotations } from '../Annotations/utilities/checkAvailableAnnotations.js';
import { XAxis } from '../XAxis/XAxis.js';
import { GradientDefs } from '../shared/GradientDefs/GradientDefs.js';
import { HorizontalGridLines } from '../HorizontalGridLines/HorizontalGridLines.js';
import { YAxis } from '../YAxis/YAxis.js';
import { VerticalBarGroup } from './components/VerticalBarGroup/VerticalBarGroup.js';
import { Annotations } from '../Annotations/Annotations.js';
import { YAxisAnnotations } from '../Annotations/YAxisAnnotations.js';
import { TooltipWrapper } from '../TooltipWrapper/TooltipWrapper.js';
import { LegendContainer } from '../LegendContainer/LegendContainer.js';
import { TOOLTIP_POSITION_DEFAULT_RETURN } from '../TooltipWrapper/constants.js';
import { TooltipHorizontalOffset, TooltipVerticalOffset } from '../TooltipWrapper/types.js';

function Chart({
  annotationsLookupTable = {},
  data,
  dimensions,
  emptyStateText,
  renderLegendContent,
  renderTooltipContent,
  showLegend,
  type,
  xAxisOptions,
  yAxisOptions,
  renderHiddenLegendLabel,
  seriesNameFormatter
}) {
  useColorVisionEvents({
    enabled: data.length > 1,
    dimensions
  });
  const selectedTheme = useTheme();
  const {
    characterWidths
  } = useChartContext();
  const [svgRef, setSvgRef] = useState(null);
  const id = useMemo(() => uniqueId('VerticalBarChart'), []);
  const [xAxisHeight, setXAxisHeight] = useState(LINE_HEIGHT);
  const [annotationsHeight, setAnnotationsHeight] = useState(0);
  const {
    legend,
    setLegendDimensions,
    height,
    width
  } = useLegend({
    data: [{
      shape: 'Bar',
      series: data
    }],
    dimensions,
    showLegend,
    seriesNameFormatter
  });
  const emptyState = data.length === 0;
  const indexForLabels = useIndexForLabels(data);
  const {
    formattedLabels,
    unformattedLabels
  } = useFormattedLabels({
    data: [data[indexForLabels]],
    labelFormatter: xAxisOptions.labelFormatter
  });
  const isStacked = type === 'stacked';
  const stackedValues = isStacked ? getStackedValues({
    series: data,
    labels: formattedLabels,
    order: stackOrderNone,
    offset: stackOffsetDiverging
  }) : null;
  const reducedLabelIndexes = useReducedLabelIndexes({
    dataLength: data[0] ? data[0].data.length : 0
  });
  const {
    min,
    max
  } = getStackedMinMax({
    stackedValues,
    data,
    integersOnly: yAxisOptions.integersOnly
  });
  let yScaleMax;

  if (yAxisOptions.maxYOverride === null) {
    yScaleMax = max;
  } else {
    const allValuesAreZero = data.every(series => series.data.every(point => point.value === 0));
    yScaleMax = allValuesAreZero ? 0 : max;
  }

  const yScaleOptions = {
    formatYAxisLabel: yAxisOptions.labelFormatter,
    integersOnly: yAxisOptions.integersOnly,
    maxYOverride: yAxisOptions.maxYOverride,
    max: yScaleMax,
    min
  };
  const {
    ticks: initialTicks
  } = useYScale({ ...yScaleOptions,
    drawableHeight: height,
    verticalOverflow: selectedTheme.grid.verticalOverflow
  });
  const yAxisLabelWidth = useMemo(() => {
    const longest = Math.max(...initialTicks.map(({
      formattedValue
    }) => estimateStringWidth(formattedValue, characterWidths)));
    return longest;
  }, [characterWidths, initialTicks]);
  const {
    drawableWidth,
    drawableHeight,
    chartXPosition,
    chartYPosition,
    xAxisBounds,
    yAxisBounds
  } = useChartPositions({
    annotationsHeight,
    height,
    width,
    xAxisHeight,
    yAxisWidth: yAxisLabelWidth
  });
  const annotationsDrawableHeight = chartYPosition + drawableHeight + ANNOTATIONS_LABELS_OFFSET;
  const chartBounds = {
    width,
    height,
    x: chartXPosition,
    y: chartYPosition
  };
  const hideXAxis = xAxisOptions.hide || selectedTheme.xAxis.hide;
  const {
    sortedData,
    areAllNegative,
    xScale,
    gapWidth
  } = useVerticalBarChart({
    data,
    drawableWidth,
    labels: formattedLabels
  });
  const {
    ticks,
    yScale
  } = useYScale({ ...yScaleOptions,
    drawableHeight,
    verticalOverflow: selectedTheme.grid.verticalOverflow
  });
  const barColors = data.map(({
    color
  }) => color);
  const getTooltipMarkup = useBarChartTooltipContent({
    renderTooltipContent,
    data,
    seriesColors: barColors,
    seriesNameFormatter
  });
  const {
    hasXAxisAnnotations,
    hasYAxisAnnotations
  } = checkAvailableAnnotations(annotationsLookupTable);
  const xAxisLabelHalf = xScale.bandwidth() / 2;
  return /*#__PURE__*/jsxs(ChartElements.Div, {
    height: height,
    width: width,
    children: [/*#__PURE__*/jsxs(ChartElements.Svg, {
      height: height,
      width: width,
      emptyStateText: emptyStateText,
      emptyState: emptyState,
      setRef: setSvgRef,
      children: [hideXAxis ? null : /*#__PURE__*/jsx(XAxis, {
        allowLineWrap: xAxisOptions.allowLineWrap,
        labels: formattedLabels,
        labelWidth: xScale.bandwidth(),
        onHeightChange: setXAxisHeight,
        reducedLabelIndexes: reducedLabelIndexes,
        x: xAxisBounds.x,
        xScale: xScale,
        y: xAxisBounds.y
      }), /*#__PURE__*/jsx(GradientDefs, {
        direction: "vertical",
        gradientUnits: isStacked ? 'objectBoundingBox' : 'userSpaceOnUse',
        id: id,
        seriesColors: barColors,
        size: isStacked ? '100%' : `${width}px`
      }), selectedTheme.grid.showHorizontalLines ? /*#__PURE__*/jsx(HorizontalGridLines, {
        ticks: ticks,
        transform: {
          x: selectedTheme.grid.horizontalOverflow ? 0 : chartXPosition,
          y: chartYPosition
        },
        width: selectedTheme.grid.horizontalOverflow ? width : drawableWidth
      }) : null, /*#__PURE__*/jsx(YAxis, {
        ticks: ticks,
        textAlign: "right",
        width: yAxisLabelWidth,
        x: yAxisBounds.x,
        y: yAxisBounds.y
      }), /*#__PURE__*/jsx("g", {
        transform: `translate(${chartXPosition},${chartYPosition})`,
        children: /*#__PURE__*/jsx(VerticalBarGroup, {
          colors: barColors,
          data: data,
          drawableHeight: drawableHeight,
          gapWidth: gapWidth,
          id: id,
          labels: formattedLabels,
          sortedData: sortedData,
          stackedValues: stackedValues,
          xScale: xScale,
          yAxisOptions: yAxisOptions,
          yScale: yScale,
          areAllNegative: areAllNegative
        })
      }), hasXAxisAnnotations && /*#__PURE__*/jsx("g", {
        transform: `translate(${chartXPosition + xAxisLabelHalf},0)`,
        tabIndex: -1,
        children: /*#__PURE__*/jsx(Annotations, {
          annotationsLookupTable: annotationsLookupTable,
          axisLabelWidth: xScale.bandwidth(),
          drawableHeight: annotationsDrawableHeight,
          drawableWidth: drawableWidth,
          labels: unformattedLabels,
          onHeightChange: setAnnotationsHeight,
          xScale: xScale
        })
      }), hasYAxisAnnotations && /*#__PURE__*/jsx("g", {
        transform: `translate(${chartXPosition},${chartYPosition})`,
        tabIndex: -1,
        children: /*#__PURE__*/jsx(YAxisAnnotations, {
          annotationsLookupTable: annotationsLookupTable,
          drawableHeight: annotationsDrawableHeight,
          drawableWidth: drawableWidth,
          ticks: ticks,
          yScale: yScale
        })
      })]
    }), sortedData.length > 0 && /*#__PURE__*/jsx(TooltipWrapper, {
      bandwidth: xScale.bandwidth(),
      chartBounds: chartBounds,
      focusElementDataType: DataType.BarGroup,
      getMarkup: getTooltipMarkup,
      getPosition: getTooltipPosition,
      margin: { ...ChartMargin,
        Top: chartYPosition
      },
      parentRef: svgRef,
      chartDimensions: dimensions,
      usePortal: true
    }), showLegend && /*#__PURE__*/jsx(LegendContainer, {
      colorVisionType: COLOR_VISION_SINGLE_ITEM,
      data: legend,
      onDimensionChange: setLegendDimensions,
      renderLegendContent: renderLegendContent,
      enableHideOverflow: true,
      dimensions: dimensions,
      renderHiddenLegendLabel: renderHiddenLegendLabel
    })]
  });

  function formatPositionForTooltip(index) {
    var _xScale, _dimensions$x, _dimensions$y;

    if (index == null) {
      return TOOLTIP_POSITION_DEFAULT_RETURN;
    }

    const xPosition = (_xScale = xScale(`${index}`)) !== null && _xScale !== void 0 ? _xScale : 0;
    const sortedDataPos = sortedData[index].map(num => Math.abs(num !== null && num !== void 0 ? num : 0));
    const highestValuePos = type === 'stacked' ? sortedData[index].reduce(sumPositiveData, 0) : Math.max(...sortedDataPos);
    const x = xPosition + chartXPosition;
    const y = yScale(highestValuePos) + chartYPosition;
    return {
      x: x + ((_dimensions$x = dimensions === null || dimensions === void 0 ? void 0 : dimensions.x) !== null && _dimensions$x !== void 0 ? _dimensions$x : 0),
      y: Math.abs(y) + ((_dimensions$y = dimensions === null || dimensions === void 0 ? void 0 : dimensions.y) !== null && _dimensions$y !== void 0 ? _dimensions$y : 0),
      position: {
        horizontal: TooltipHorizontalOffset.Center,
        vertical: areAllNegative ? TooltipVerticalOffset.Below : TooltipVerticalOffset.Above
      },
      activeIndex: index
    };
  }

  function getTooltipPosition({
    event,
    index,
    eventType
  }) {
    return getVerticalBarChartTooltipPosition({
      tooltipPosition: {
        event,
        index,
        eventType
      },
      chartXPosition,
      formatPositionForTooltip,
      maxIndex: sortedData.length - 1,
      step: xScale.step(),
      yMin: ChartMargin.Top,
      yMax: drawableHeight + Number(ChartMargin.Bottom) + xAxisHeight
    });
  }
}

function sumPositiveData(prevValue, currValue) {
  return currValue < 0 ? prevValue : prevValue + currValue;
}

export { Chart };
