// @shopify/polaris-viz/build/cjs/components/StackedAreaChart/components/Area/Area.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var Area$1 = require('./Area.scss.js');
var jsxRuntime = require('react/jsx-runtime');

function Area({
  activeLineIndex,
  areaGenerator,
  colors,
  data,
  id,
  index,
  lineGenerator,
  selectedTheme
}) {
  const opacity = 0.25;
  const areaShape = areaGenerator(data);
  const lineShape = lineGenerator(data);

  if (areaShape == null || lineShape == null) {
    return null;
  }

  const gradient = polarisVizCore.getGradientFromColor(colors[index]);
  return /*#__PURE__*/jsxRuntime.jsxs("g", { ...polarisVizCore.getColorVisionEventAttrs({
      type: polarisVizCore.COLOR_VISION_SINGLE_ITEM,
      index
    }),
    tabIndex: -1,
    children: [/*#__PURE__*/jsxRuntime.jsx("defs", {
      children: /*#__PURE__*/jsxRuntime.jsx(polarisVizCore.LinearGradientWithStops, {
        id: `area-${id}-${index}`,
        gradient: gradient,
        gradientUnits: "userSpaceOnUse",
        y1: "100%",
        y2: "0%"
      })
    }), /*#__PURE__*/jsxRuntime.jsxs("g", {
      style: polarisVizCore.getColorVisionStylesForActiveIndex({
        activeIndex: activeLineIndex,
        index
      }),
      "aria-hidden": "true",
      tabIndex: -1,
      className: Area$1["default"].Group,
      children: [/*#__PURE__*/jsxRuntime.jsx("path", {
        d: lineShape,
        fill: "none",
        stroke: `url(#area-${id}-${index})`,
        strokeWidth: selectedTheme.line.width
      }, `line-${index}`), /*#__PURE__*/jsxRuntime.jsx("path", {
        style: {
          opacity
        },
        d: areaShape,
        fill: `url(#area-${id}-${index})`
      }, index)]
    })]
  });
}

exports.Area = Area;
