// @shopify/polaris-viz/build/esm/components/SimpleBarChart/SimpleBarChart.js
import { usePolarisVizContext, DEFAULT_CHART_PROPS, ChartState } from '@shopify/polaris-viz-core';
import { Chart } from './Chart.js';
import { jsx } from 'react/jsx-runtime';
import { getXAxisOptionsWithDefaults, getYAxisOptionsWithDefaults } from '../../utilities/getAxisOptions.js';
import { ChartContainer } from '../ChartContainer/ChartContainer.js';
import { ChartSkeleton } from '../ChartSkeleton/ChartSkeleton.js';

function SimpleBarChart(props) {
  const {
    defaultTheme
  } = usePolarisVizContext();
  const {
    id,
    isAnimated,
    data,
    renderLegendContent,
    legendPosition = 'bottom-right',
    onError,
    seriesNameFormatter = value => `${value}`,
    showLegend = true,
    theme = defaultTheme,
    type = 'default',
    xAxisOptions,
    yAxisOptions,
    state,
    errorText
  } = { ...DEFAULT_CHART_PROPS,
    ...props
  };
  const xAxisOptionsWithDefaults = getXAxisOptionsWithDefaults(xAxisOptions);
  const yAxisOptionsWithDefaults = getYAxisOptionsWithDefaults(yAxisOptions);
  return /*#__PURE__*/jsx(ChartContainer, {
    data: data,
    theme: theme,
    id: id,
    isAnimated: isAnimated,
    onError: onError,
    children: state !== ChartState.Success ? /*#__PURE__*/jsx(ChartSkeleton, {
      type: "SimpleBar",
      state: state,
      errorText: errorText,
      theme: theme
    }) : /*#__PURE__*/jsx(Chart, {
      data: data,
      renderLegendContent: renderLegendContent,
      seriesNameFormatter: seriesNameFormatter,
      legendPosition: legendPosition,
      showLegend: showLegend,
      type: type,
      xAxisOptions: xAxisOptionsWithDefaults,
      yAxisOptions: yAxisOptionsWithDefaults
    })
  });
}

export { SimpleBarChart };
