// @shopify/polaris-viz/build/cjs/components/shared/GradientDefs/GradientDefs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var jsxRuntime = require('react/jsx-runtime');

const GRADIENT_ID = 'grad';
function GradientDefs({
  direction = 'vertical',
  id,
  seriesColors = [],
  size = '100%',
  gradientUnits
}) {
  const {
    theme
  } = polarisVizCore.useChartContext();
  return /*#__PURE__*/jsxRuntime.jsx("defs", {
    children: seriesColors.map((color, index) => {
      const gradId = getGradientDefId(theme, index, id);
      return /*#__PURE__*/jsxRuntime.jsx(Gradient, {
        color: color,
        direction: direction,
        gradientUnits: gradientUnits,
        id: gradId,
        size: size
      }, gradId);
    })
  });
}

function Gradient({
  id,
  color,
  direction,
  size,
  gradientUnits = 'userSpaceOnUse'
}) {
  const gradient = polarisVizCore.getGradientFromColor(color);
  const position = direction === 'vertical' ? {
    y1: size
  } : {
    x2: size
  };
  return /*#__PURE__*/jsxRuntime.jsx(polarisVizCore.LinearGradientWithStops, {
    gradient: gradient,
    gradientUnits: gradientUnits,
    id: id,
    ...position
  });
}

function getGradientDefId(theme = polarisVizCore.DEFAULT_THEME_NAME, index, id) {
  return [id, theme, index, GRADIENT_ID].join('-');
}

exports.GradientDefs = GradientDefs;
exports.getGradientDefId = getGradientDefId;
