// @shopify/polaris-viz/build/cjs/components/Annotations/hooks/useAnnotationPositions.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var polarisVizCore = require('@shopify/polaris-viz-core');
var constants$1 = require('../../../constants.js');
var constants = require('../constants.js');
var checkForHorizontalSpace = require('../utilities/checkForHorizontalSpace.js');
var useShowMoreAnnotationsButton = require('./useShowMoreAnnotationsButton.js');

function useAnnotationPositions({
  annotations,
  axisLabelWidth,
  dataIndexes,
  drawableWidth,
  isShowingAllAnnotations,
  onHeightChange,
  xScale
}) {
  const {
    characterWidths
  } = polarisVizCore.useChartContext();
  const textWidths = React.useMemo(() => {
    return annotations.map(annotation => {
      return polarisVizCore.estimateStringWidth(annotation.label, characterWidths);
    });
  }, [annotations, characterWidths]);
  const {
    positions,
    hiddenAnnotationsCount
  } = React.useMemo(() => {
    let positions = annotations.map((annotation, dataIndex) => {
      const xPosition = polarisVizCore.getValueFromXScale(dataIndexes[annotation.startKey], xScale);
      const textWidth = textWidths[dataIndex];
      const width = polarisVizCore.clamp({
        amount: textWidth + constants.PILL_PADDING * 2,
        min: textWidth + constants.PILL_PADDING * 2,
        max: drawableWidth
      });
      const rawX = polarisVizCore.clamp({
        amount: xPosition,
        min: xPosition,
        max: xPosition + axisLabelWidth
      });
      let x = polarisVizCore.clamp({
        amount: rawX - width / 2,
        min: constants.PILL_X_MIN,
        max: Infinity
      });
      const right = x + width;

      if (right > drawableWidth) {
        x -= right - drawableWidth;
      }

      return {
        index: dataIndex,
        line: {
          x: rawX,
          y: 0
        },
        row: 1,
        width,
        x,
        y: 0
      };
    });
    positions = positions.sort((one, two) => one.x - two.x);
    checkForHorizontalSpace.checkForHorizontalSpace({
      positions,
      totalRows: 1
    });
    positions.forEach(current => {
      const row = current.row - 1;
      current.y = row * constants.PILL_HEIGHT + row * constants.PILL_ROW_GAP;
    });
    const hiddenAnnotationsCount = positions.filter(({
      row
    }) => row >= constants$1.COLLAPSED_ANNOTATIONS_COUNT).length;
    return {
      positions,
      hiddenAnnotationsCount
    };
  }, [annotations, dataIndexes, textWidths, axisLabelWidth, xScale, drawableWidth]);
  const {
    totalRowHeight,
    rowCount
  } = useShowMoreAnnotationsButton.useShowMoreAnnotationsButton({
    isShowingAllAnnotations,
    positions
  });
  React.useEffect(() => {
    onHeightChange(totalRowHeight);
  }, [onHeightChange, totalRowHeight]);
  return {
    positions,
    rowCount,
    hiddenAnnotationsCount: isShowingAllAnnotations ? 0 : hiddenAnnotationsCount
  };
}

exports.useAnnotationPositions = useAnnotationPositions;
