// @shopify/polaris-viz/build/esm/components/SimpleNormalizedChart/components/BarLabel/BarLabel.js
import { useTheme, getColorVisionStylesForActiveIndex, getColorVisionEventAttrs, COLOR_VISION_SINGLE_ITEM } from '@shopify/polaris-viz-core';
import { getCSSBackgroundFromColor } from '../../../../utilities/getCSSBackgroundFromColor.js';
import styles from './BarLabel.scss.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { ComparisonMetric } from '../../../ComparisonMetric/ComparisonMetric.js';
import { classNames } from '../../../../utilities/classnames.js';

function BarLabel({
  activeIndex,
  index,
  label,
  value,
  color,
  comparisonMetric,
  direction,
  legendPosition
}) {
  const selectedTheme = useTheme();
  const {
    labelColor,
    valueColor
  } = selectedTheme.legend;
  const comparisonIndicator = comparisonMetric ? /*#__PURE__*/jsx(ComparisonMetric, { ...comparisonMetric
  }) : null;
  const angle = direction === 'horizontal' ? 90 : 180;
  const backgroundColor = getCSSBackgroundFromColor(color, angle);
  return /*#__PURE__*/jsxs("li", {
    className: classNames(styles.Container, legendPosition.includes('bottom') ? styles.ContaineBottomLabel : styles.ContainerDefaultLabel),
    style: getColorVisionStylesForActiveIndex({
      activeIndex,
      index
    }),
    ...getColorVisionEventAttrs({
      type: COLOR_VISION_SINGLE_ITEM,
      index
    }),
    children: [/*#__PURE__*/jsx("div", {
      style: {
        background: backgroundColor
      },
      className: styles.LabelColor
    }), /*#__PURE__*/jsxs("div", {
      className: styles.Label,
      children: [/*#__PURE__*/jsx("div", {
        style: {
          color: labelColor
        },
        className: direction === 'horizontal' ? styles.FormattedHorizontalLabel : styles.FormattedVerticalLabel,
        children: label
      }), /*#__PURE__*/jsxs("div", {
        style: {
          color: valueColor
        },
        className: direction === 'horizontal' ? styles.ValueHorizontalContainer : styles.ValueContainer,
        children: [/*#__PURE__*/jsx("span", {
          className: styles.Value,
          children: value
        }), comparisonIndicator]
      })]
    })]
  });
}

export { BarLabel };
