// @shopify/polaris-viz/build/cjs/components/HorizontalBarChartXAxisLabels/HorizontalBarChartXAxisLabels.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var jsxRuntime = require('react/jsx-runtime');
var useLabels = require('../Labels/hooks/useLabels.js');
var TextLine = require('../TextLine/TextLine.js');

function HorizontalBarChartXAxisLabels({
  allowLineWrap,
  chartX,
  chartY,
  labels,
  labelWidth,
  onHeightChange,
  ticks,
  xScale
}) {
  const {
    lines
  } = useLabels.useLabels({
    allowLineWrap,
    labels,
    onHeightChange,
    targetWidth: labelWidth
  });
  return /*#__PURE__*/jsxRuntime.jsx(React.Fragment, {
    children: lines.map((line, index) => {
      var _xScale;

      const x = (_xScale = xScale(ticks[index])) !== null && _xScale !== void 0 ? _xScale : 0;
      return /*#__PURE__*/jsxRuntime.jsx("g", {
        transform: `translate(${chartX + x},${chartY})`,
        children: /*#__PURE__*/jsxRuntime.jsx(TextLine.TextLine, {
          line: line,
          index: index
        })
      }, index);
    })
  });
}

exports.HorizontalBarChartXAxisLabels = HorizontalBarChartXAxisLabels;
