// @shopify/polaris-viz/build/esm/components/ComboChart/components/AxisLabel/AxisLabel.js
import { useTheme, LINE_HEIGHT } from '@shopify/polaris-viz-core';
import { useEstimateStringWidth } from '../../../../hooks/useEstimateStringWidth.js';
import { jsxs, jsx } from 'react/jsx-runtime';
import { SingleTextLine } from '../../../Labels/SingleTextLine.js';

function AxisLabel({
  height,
  name,
  axis,
  x,
  y
}) {
  const stringWidth = useEstimateStringWidth(name);
  const selectedTheme = useTheme();
  const rotate = axis === 'primary' ? -90 : 90;
  const yCenter = height / 2;
  const halfStringSize = stringWidth / 2;
  const yOffset = axis === 'primary' ? yCenter : yCenter - stringWidth;
  const xOffset = axis === 'primary' ? 0 : LINE_HEIGHT;
  return /*#__PURE__*/jsxs("g", {
    transform: `translate(${x + xOffset},${y + yOffset + halfStringSize}) rotate(${rotate})`,
    children: [/*#__PURE__*/jsx("rect", {
      width: stringWidth,
      height: LINE_HEIGHT,
      fill: selectedTheme.chartContainer.backgroundColor
    }), /*#__PURE__*/jsx(SingleTextLine, {
      color: selectedTheme.yAxis.labelColor,
      targetWidth: Math.max(stringWidth, height),
      text: name,
      x: 0,
      y: 0
    })]
  });
}

export { AxisLabel };
