// @shopify/polaris-viz-core/build/cjs/types.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

// === Theme types === //
exports.DataType = void 0;

(function (DataType) {
  DataType["Point"] = "Point";
  DataType["BarGroup"] = "BarGroup";
  DataType["Bar"] = "Bar";
})(exports.DataType || (exports.DataType = {}));

exports.ChartState = void 0;

(function (ChartState) {
  ChartState["Loading"] = "Loading";
  ChartState["Error"] = "Error";
  ChartState["Success"] = "Success";
})(exports.ChartState || (exports.ChartState = {}));

exports.InternalChartType = void 0;

(function (InternalChartType) {
  InternalChartType["Bar"] = "Bar";
  InternalChartType["Combo"] = "Combo";
  InternalChartType["Line"] = "Line";
})(exports.InternalChartType || (exports.InternalChartType = {}));

exports.Hue = void 0;

(function (Hue) {
  Hue["Teal"] = "Teal";
  Hue["Blue"] = "Blue";
  Hue["Indigo"] = "Indigo";
  Hue["Purple"] = "Purple";
  Hue["Magenta"] = "Magenta";
  Hue["Orange"] = "Orange";
  Hue["Yellow"] = "Yellow";
})(exports.Hue || (exports.Hue = {}));
