// @shopify/polaris-viz/build/cjs/components/LineChartPredictive/components/CustomLegend/CustomLegend.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var CustomLegend$1 = require('./CustomLegend.scss.js');
var jsxRuntime = require('react/jsx-runtime');
var LegendItem = require('../../../Legend/components/LegendItem/LegendItem.js');
var SeriesIcon = require('../SeriesIcon/SeriesIcon.js');

function CustomLegend({
  data,
  predictiveSeriesNames,
  getColorVisionEventAttrs,
  getColorVisionStyles,
  seriesNameFormatter,
  theme
}) {
  return /*#__PURE__*/jsxRuntime.jsx("ul", {
    className: CustomLegend$1["default"].Container,
    children: data.map(({
      color,
      name,
      isComparison,
      metadata
    }, index) => {
      if (metadata !== null && metadata !== void 0 && metadata.isPredictive) {
        return null;
      }

      function renderSeriesIcon() {
        return /*#__PURE__*/jsxRuntime.jsx(SeriesIcon.SeriesIcon, {
          color: color
        });
      }

      return /*#__PURE__*/jsxRuntime.jsx("li", {
        style: { ...getColorVisionStyles(index)
        },
        ...getColorVisionEventAttrs(index),
        children: /*#__PURE__*/jsxRuntime.jsx(LegendItem.LegendItem, {
          color: color,
          index: index,
          renderSeriesIcon: predictiveSeriesNames.includes(name !== null && name !== void 0 ? name : '') ? renderSeriesIcon : undefined,
          isComparison: isComparison,
          name: seriesNameFormatter(name !== null && name !== void 0 ? name : ''),
          shape: "Line",
          theme: theme
        })
      }, index);
    })
  });
}

exports.CustomLegend = CustomLegend;
