// @shopify/polaris-viz/build/cjs/constants.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');

const DEFAULT_LEGEND_HEIGHT = 24;
const DEFAULT_LEGEND_WIDTH = 29;
const BAR_CONTAINER_TEXT_HEIGHT = 48;
const ANNOTATIONS_LABELS_OFFSET = 10;
const Y_AXIS_LABEL_OFFSET = 2;
const TOOLTIP_BG_OPACITY = 0.8;
const COLLAPSED_ANNOTATIONS_COUNT = 3;
const PREVIEW_ICON_SIZE = 12;
const ARC_PAD_ANGLE = 0.02;
const ZERO_VALUE_LINE_HEIGHT = 6;
const NEGATIVE_ZERO_LINE_OFFSET = 10;
const IS_DEVELOPMENT = process.env.NODE_ENV === 'development';
const IS_TEST = process.env.NODE_ENV === 'test';
const WARN_FOR_DEVELOPMENT = IS_DEVELOPMENT && !IS_TEST;
const HOVER_TARGET_ZONE = 48;
const CROSSHAIR_ID = 'Crosshair';
const DEFAULT_ANIMATION_DELAY = 100;

Object.defineProperty(exports, 'BARS_LOAD_ANIMATION_CONFIG', {
  enumerable: true,
  get: function () { return polarisVizCore.BARS_LOAD_ANIMATION_CONFIG; }
});
Object.defineProperty(exports, 'BARS_SORT_TRANSITION_CONFIG', {
  enumerable: true,
  get: function () { return polarisVizCore.BARS_SORT_TRANSITION_CONFIG; }
});
Object.defineProperty(exports, 'BARS_TRANSITION_CONFIG', {
  enumerable: true,
  get: function () { return polarisVizCore.BARS_TRANSITION_CONFIG; }
});
Object.defineProperty(exports, 'BASE_ANIMATION_DURATION', {
  enumerable: true,
  get: function () { return polarisVizCore.BASE_ANIMATION_DURATION; }
});
Object.defineProperty(exports, 'ChartMargin', {
  enumerable: true,
  get: function () { return polarisVizCore.ChartMargin; }
});
Object.defineProperty(exports, 'DARK_THEME', {
  enumerable: true,
  get: function () { return polarisVizCore.DARK_THEME; }
});
Object.defineProperty(exports, 'DEFAULT_BORDER_RADIUS', {
  enumerable: true,
  get: function () { return polarisVizCore.DEFAULT_BORDER_RADIUS; }
});
Object.defineProperty(exports, 'DEFAULT_MAX_Y', {
  enumerable: true,
  get: function () { return polarisVizCore.DEFAULT_MAX_Y; }
});
Object.defineProperty(exports, 'DIAGONAL_LABEL_MIN_WIDTH', {
  enumerable: true,
  get: function () { return polarisVizCore.DIAGONAL_LABEL_MIN_WIDTH; }
});
Object.defineProperty(exports, 'ELLIPSIS', {
  enumerable: true,
  get: function () { return polarisVizCore.ELLIPSIS; }
});
Object.defineProperty(exports, 'EMPTY_STATE_CHART_MAX', {
  enumerable: true,
  get: function () { return polarisVizCore.EMPTY_STATE_CHART_MAX; }
});
Object.defineProperty(exports, 'EMPTY_STATE_CHART_MIN', {
  enumerable: true,
  get: function () { return polarisVizCore.EMPTY_STATE_CHART_MIN; }
});
Object.defineProperty(exports, 'FONT_SIZE', {
  enumerable: true,
  get: function () { return polarisVizCore.FONT_SIZE; }
});
Object.defineProperty(exports, 'HORIZONTAL_BAR_LABEL_HEIGHT', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_BAR_LABEL_HEIGHT; }
});
Object.defineProperty(exports, 'HORIZONTAL_BAR_LABEL_OFFSET', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_BAR_LABEL_OFFSET; }
});
Object.defineProperty(exports, 'HORIZONTAL_GROUP_LABEL_HEIGHT', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_GROUP_LABEL_HEIGHT; }
});
Object.defineProperty(exports, 'HORIZONTAL_LABEL_MIN_WIDTH', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_LABEL_MIN_WIDTH; }
});
Object.defineProperty(exports, 'HORIZONTAL_LABEL_TARGET_HEIGHT', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_LABEL_TARGET_HEIGHT; }
});
Object.defineProperty(exports, 'HORIZONTAL_SPACE_BETWEEN_CHART_AND_AXIS', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_SPACE_BETWEEN_CHART_AND_AXIS; }
});
Object.defineProperty(exports, 'HORIZONTAL_SPACE_BETWEEN_SETS', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_SPACE_BETWEEN_SETS; }
});
Object.defineProperty(exports, 'HORIZONTAL_SPACE_BETWEEN_SINGLE', {
  enumerable: true,
  get: function () { return polarisVizCore.HORIZONTAL_SPACE_BETWEEN_SINGLE; }
});
Object.defineProperty(exports, 'LABEL_AREA_TOP_SPACING', {
  enumerable: true,
  get: function () { return polarisVizCore.LABEL_AREA_TOP_SPACING; }
});
Object.defineProperty(exports, 'LIGHT_THEME', {
  enumerable: true,
  get: function () { return polarisVizCore.LIGHT_THEME; }
});
Object.defineProperty(exports, 'LINES_LOAD_ANIMATION_CONFIG', {
  enumerable: true,
  get: function () { return polarisVizCore.LINES_LOAD_ANIMATION_CONFIG; }
});
Object.defineProperty(exports, 'LINE_ANIMATION_DURATION_STEP', {
  enumerable: true,
  get: function () { return polarisVizCore.LINE_ANIMATION_DURATION_STEP; }
});
Object.defineProperty(exports, 'LINE_ANIMATION_FAST_COUNT', {
  enumerable: true,
  get: function () { return polarisVizCore.LINE_ANIMATION_FAST_COUNT; }
});
Object.defineProperty(exports, 'LINE_ANIMATION_FAST_DURATION', {
  enumerable: true,
  get: function () { return polarisVizCore.LINE_ANIMATION_FAST_DURATION; }
});
Object.defineProperty(exports, 'LINE_ANIMATION_SLOW_DURATION', {
  enumerable: true,
  get: function () { return polarisVizCore.LINE_ANIMATION_SLOW_DURATION; }
});
Object.defineProperty(exports, 'LINE_HEIGHT', {
  enumerable: true,
  get: function () { return polarisVizCore.LINE_HEIGHT; }
});
Object.defineProperty(exports, 'LOAD_ANIMATION_DURATION', {
  enumerable: true,
  get: function () { return polarisVizCore.LOAD_ANIMATION_DURATION; }
});
Object.defineProperty(exports, 'MASK_HIGHLIGHT_COLOR', {
  enumerable: true,
  get: function () { return polarisVizCore.MASK_HIGHLIGHT_COLOR; }
});
Object.defineProperty(exports, 'MAX_DIAGONAL_LABEL_WIDTH', {
  enumerable: true,
  get: function () { return polarisVizCore.MAX_DIAGONAL_LABEL_WIDTH; }
});
Object.defineProperty(exports, 'MAX_DIAGONAL_VISIBLE_HEIGHT', {
  enumerable: true,
  get: function () { return polarisVizCore.MAX_DIAGONAL_VISIBLE_HEIGHT; }
});
Object.defineProperty(exports, 'MIN_BAR_HEIGHT', {
  enumerable: true,
  get: function () { return polarisVizCore.MIN_BAR_HEIGHT; }
});
Object.defineProperty(exports, 'NEUTRAL_SINGLE_GRADIENT', {
  enumerable: true,
  get: function () { return polarisVizCore.NEUTRAL_SINGLE_GRADIENT; }
});
Object.defineProperty(exports, 'PRINT_THEME', {
  enumerable: true,
  get: function () { return polarisVizCore.PRINT_THEME; }
});
Object.defineProperty(exports, 'SHAPE_ANIMATION_HEIGHT_BUFFER', {
  enumerable: true,
  get: function () { return polarisVizCore.SHAPE_ANIMATION_HEIGHT_BUFFER; }
});
Object.defineProperty(exports, 'STACKED_BAR_GAP', {
  enumerable: true,
  get: function () { return polarisVizCore.STACKED_BAR_GAP; }
});
Object.defineProperty(exports, 'VERTICAL_LABEL_MIN_WIDTH', {
  enumerable: true,
  get: function () { return polarisVizCore.VERTICAL_LABEL_MIN_WIDTH; }
});
Object.defineProperty(exports, 'VERTICAL_LABEL_TARGET_WIDTH', {
  enumerable: true,
  get: function () { return polarisVizCore.VERTICAL_LABEL_TARGET_WIDTH; }
});
Object.defineProperty(exports, 'XMLNS', {
  enumerable: true,
  get: function () { return polarisVizCore.XMLNS; }
});
Object.defineProperty(exports, 'Y_AXIS_CHART_SPACING', {
  enumerable: true,
  get: function () { return polarisVizCore.Y_AXIS_CHART_SPACING; }
});
exports.ANNOTATIONS_LABELS_OFFSET = ANNOTATIONS_LABELS_OFFSET;
exports.ARC_PAD_ANGLE = ARC_PAD_ANGLE;
exports.BAR_CONTAINER_TEXT_HEIGHT = BAR_CONTAINER_TEXT_HEIGHT;
exports.COLLAPSED_ANNOTATIONS_COUNT = COLLAPSED_ANNOTATIONS_COUNT;
exports.CROSSHAIR_ID = CROSSHAIR_ID;
exports.DEFAULT_ANIMATION_DELAY = DEFAULT_ANIMATION_DELAY;
exports.DEFAULT_LEGEND_HEIGHT = DEFAULT_LEGEND_HEIGHT;
exports.DEFAULT_LEGEND_WIDTH = DEFAULT_LEGEND_WIDTH;
exports.HOVER_TARGET_ZONE = HOVER_TARGET_ZONE;
exports.IS_DEVELOPMENT = IS_DEVELOPMENT;
exports.IS_TEST = IS_TEST;
exports.NEGATIVE_ZERO_LINE_OFFSET = NEGATIVE_ZERO_LINE_OFFSET;
exports.PREVIEW_ICON_SIZE = PREVIEW_ICON_SIZE;
exports.TOOLTIP_BG_OPACITY = TOOLTIP_BG_OPACITY;
exports.WARN_FOR_DEVELOPMENT = WARN_FOR_DEVELOPMENT;
exports.Y_AXIS_LABEL_OFFSET = Y_AXIS_LABEL_OFFSET;
exports.ZERO_VALUE_LINE_HEIGHT = ZERO_VALUE_LINE_HEIGHT;
