// @shopify/polaris-viz/build/cjs/components/ComboChart/components/AxisLabel/AxisLabel.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var polarisVizCore = require('@shopify/polaris-viz-core');
var useEstimateStringWidth = require('../../../../hooks/useEstimateStringWidth.js');
var jsxRuntime = require('react/jsx-runtime');
var SingleTextLine = require('../../../Labels/SingleTextLine.js');

function AxisLabel({
  height,
  name,
  axis,
  x,
  y
}) {
  const stringWidth = useEstimateStringWidth.useEstimateStringWidth(name);
  const selectedTheme = polarisVizCore.useTheme();
  const rotate = axis === 'primary' ? -90 : 90;
  const yCenter = height / 2;
  const halfStringSize = stringWidth / 2;
  const yOffset = axis === 'primary' ? yCenter : yCenter - stringWidth;
  const xOffset = axis === 'primary' ? 0 : polarisVizCore.LINE_HEIGHT;
  return /*#__PURE__*/jsxRuntime.jsxs("g", {
    transform: `translate(${x + xOffset},${y + yOffset + halfStringSize}) rotate(${rotate})`,
    children: [/*#__PURE__*/jsxRuntime.jsx("rect", {
      width: stringWidth,
      height: polarisVizCore.LINE_HEIGHT,
      fill: selectedTheme.chartContainer.backgroundColor
    }), /*#__PURE__*/jsxRuntime.jsx(SingleTextLine.SingleTextLine, {
      color: selectedTheme.yAxis.labelColor,
      targetWidth: Math.max(stringWidth, height),
      text: name,
      x: 0,
      y: 0
    })]
  });
}

exports.AxisLabel = AxisLabel;
