// @shopify/polaris-viz-core/build/cjs/components/SparkBarSeries/SparkBarSeries.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var react = require('react');
var constants = require('../../constants.js');
var jsxRuntime = require('react/jsx-runtime');
var useTheme = require('../../hooks/useTheme.js');
var getSeriesColors = require('../../utilities/getSeriesColors.js');
var usePolarisVizContext = require('../../hooks/usePolarisVizContext.js');
var uniqueId = require('../../utilities/uniqueId.js');
var useSparkBar = require('../../hooks/useSparkBar.js');
var getAnimationTrail = require('../../utilities/getAnimationTrail.js');
var LinearGradientWithStops = require('../LinearGradientWithStops/LinearGradientWithStops.js');
var Bar = require('../Bar/Bar.js');

function SparkBarSeries({
  data,
  targetLine,
  height,
  shouldAnimate,
  useTransition,
  width,
  theme
}) {
  const selectedTheme = useTheme.useTheme(theme);
  const [seriesColor] = getSeriesColors.getSeriesColors(1, selectedTheme);
  const {
    // eslint-disable-next-line id-length
    components: {
      Defs,
      Mask,
      G,
      Rect,
      Line
    },
    animated
  } = usePolarisVizContext.usePolarisVizContext();
  const AnimatedG = animated(G);
  const id = react.useMemo(() => uniqueId.uniqueId('sparkbar-series'), []);
  const clipId = react.useMemo(() => uniqueId.uniqueId('sparkbar-series-clip'), []);
  const {
    dataWithIndex,
    color,
    getBarHeight,
    strokeDashoffset,
    xScale,
    yScale,
    barWidth,
    targetLineYPosition,
    targetLineX1,
    targetLineX2
  } = useSparkBar.useSparkBar({
    data,
    height,
    width,
    seriesColor,
    targetLine
  });
  const transitions = useTransition(dataWithIndex, {
    key: ({
      index
    }) => index,
    from: {
      height: 0
    },
    leave: {
      height: 0
    },
    enter: ({
      value: {
        value
      }
    }) => ({
      height: getBarHeight(value == null ? 0 : value)
    }),
    update: ({
      value: {
        value
      }
    }) => ({
      height: getBarHeight(value == null ? 0 : value)
    }),
    default: {
      immediate: !shouldAnimate
    },
    trail: shouldAnimate ? getAnimationTrail.getAnimationTrail(dataWithIndex.length) : 0,
    config: constants.BARS_TRANSITION_CONFIG
  });
  const hasTargetLine = targetLine != null && targetLine.value != null;
  return /*#__PURE__*/jsxRuntime.jsxs(react.Fragment, {
    children: [/*#__PURE__*/jsxRuntime.jsx(Defs, {
      children: /*#__PURE__*/jsxRuntime.jsx(LinearGradientWithStops.LinearGradientWithStops, {
        id: id,
        gradient: color,
        gradientUnits: "userSpaceOnUse",
        y1: "100%",
        y2: "0%"
      })
    }), /*#__PURE__*/jsxRuntime.jsx(Mask, {
      id: clipId,
      children: /*#__PURE__*/jsxRuntime.jsx(AnimatedG, {
        opacity: hasTargetLine ? '0.9' : '1',
        children: transitions(({
          height: barHeight
        }, item, _transition, index) => {
          var _item$value$value;

          const xPosition = xScale(index.toString());
          const height = shouldAnimate ? barHeight : getBarHeight((_item$value$value = item.value.value) !== null && _item$value$value !== void 0 ? _item$value$value : 0);
          return /*#__PURE__*/jsxRuntime.jsx(Bar.Bar, {
            borderRadius: selectedTheme.bar.borderRadius,
            x: xPosition == null ? 0 : xPosition,
            yScale: yScale,
            value: item.value.value,
            width: barWidth,
            height: height,
            fill: "white"
          }, index);
        })
      })
    }), /*#__PURE__*/jsxRuntime.jsx(Rect, {
      fill: `url(#${id})`,
      width: width,
      height: height,
      mask: `url(#${clipId})`
    }), hasTargetLine ? /*#__PURE__*/jsxRuntime.jsx(Line, {
      stroke: selectedTheme.seriesColors.comparison,
      strokeWidth: constants.STROKE_WIDTH,
      x1: targetLineX1,
      x2: targetLineX2,
      y1: targetLineYPosition,
      y2: targetLineYPosition,
      strokeLinecap: "round",
      opacity: "0.9",
      strokeDashoffset: strokeDashoffset,
      strokeDasharray: constants.STROKE_DOT_ARRAY_WIDTH
    }) : null]
  });
}

exports.SparkBarSeries = SparkBarSeries;
