// @shopify/polaris-viz/build/esm/components/LineChartRelational/components/RelatedAreas/RelatedAreas.js
import { useState } from 'react';
import { useChartContext, curveStepRounded } from '@shopify/polaris-viz-core';
import { area } from 'd3-shape';
import { jsx } from 'react/jsx-runtime';
import { useExternalHideEvents } from '../../../../hooks/ExternalEvents/useExternalHideEvents.js';
import { useWatchActiveSeries } from '../../../../hooks/useWatchActiveSeries.js';
import { Area } from '../Area/Area.js';

function RelatedAreas({
  yScale,
  xScale,
  data
}) {
  const [activeIndex, setActiveIndex] = useState(-1);
  const lineSeries = data.filter(series => {
    var _series$metadata;

    return (series === null || series === void 0 ? void 0 : (_series$metadata = series.metadata) === null || _series$metadata === void 0 ? void 0 : _series$metadata.relatedIndex) == null;
  });
  const percentileIndex = lineSeries.length + 1;
  const {
    hiddenIndexes
  } = useExternalHideEvents();
  const {
    id
  } = useChartContext();
  useWatchActiveSeries(id !== null && id !== void 0 ? id : '', ({
    detail: {
      index
    }
  }) => {
    setActiveIndex(index);
  });

  function getAreaGenerator(series) {
    var _series$metadata2;

    const relatedIndex = (_series$metadata2 = series.metadata) === null || _series$metadata2 === void 0 ? void 0 : _series$metadata2.relatedIndex;
    const areaGenerator = area().x((_, index) => {
      return xScale(index);
    }).y0(({
      value
    }) => {
      return yScale(value !== null && value !== void 0 ? value : 0);
    }).y1((_, index) => {
      var _data$relatedIndex$da;

      if (data[relatedIndex] == null) {
        return yScale(0);
      }

      return yScale((_data$relatedIndex$da = data[relatedIndex].data[index].value) !== null && _data$relatedIndex$da !== void 0 ? _data$relatedIndex$da : 0);
    }).defined(({
      value
    }) => value != null).curve(curveStepRounded);
    return areaGenerator(series.data);
  }

  return /*#__PURE__*/jsx("g", {
    children: data.map((series, index) => {
      var _series$metadata3, _series$metadata4, _series$metadata5;

      if (((_series$metadata3 = series.metadata) === null || _series$metadata3 === void 0 ? void 0 : _series$metadata3.relatedIndex) == null || ((_series$metadata4 = series.metadata) === null || _series$metadata4 === void 0 ? void 0 : _series$metadata4.areaColor) == null) {
        return null;
      }

      return /*#__PURE__*/jsx(Area, {
        activeIndex: activeIndex,
        fill: (_series$metadata5 = series.metadata) === null || _series$metadata5 === void 0 ? void 0 : _series$metadata5.areaColor,
        getAreaGenerator: getAreaGenerator,
        hiddenIndexes: hiddenIndexes,
        index: percentileIndex,
        series: series,
        shouldAnimate: false
      }, index);
    })
  });
}

export { RelatedAreas };
